{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Jupyter Notebook UI to graph your TAP data!\n",
    "\n",
    "Version 1.7 - Joseph Liang, Rankin Lab\n",
    "Updated:\n",
    "1. Upgraded folder path selection application\n",
    "2. Upgraded dataset management (less moving parts for end-user)\n",
    "3. output changed from tif -> png\n",
    "4. Added tap-correction function for missing taps (should be working now)\n",
    "5. Added plate column to evaluate plate-discrepancies\n",
    "6. Added separate graphing function for plate-discrepancies\n",
    "7. Added colour palette-setting function\n",
    "\n",
    "## Known bug: Step 2 an empty windows displays in Mac. May also apply to linux/windows.\n",
    "\n",
    "## Beginner Essentials:\n",
    "1. Shift-Enter to run each cell. After you run, you should see an output \"done step #\". If not, an error has occured\n",
    "2. When inputting your own code/revising the code, make sure you close all your quotation marks '' and brackets (), [], {}.\n",
    "3. Don't leave any commas (,) hanging! (make sure an object always follows a comma. If there is nothing after a comma, remove the comma!\n",
    "4. Learning to code? Each line of code is annotated to help you understand how this code works!\n",
    "\n",
    "## 3. Run all cells/steps sequentially, even the ones that do not need input\n",
    "\n",
    "## Steps that require input: #3, #6.1, #7"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 1. Importing Packages Required (No input required, just run)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "done step 1\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd #<- package used to import and organize data\n",
    "import numpy as np #<- package used to import and organize data\n",
    "import math\n",
    "import os #<- package used to work with system file paths\n",
    "import seaborn as sns #<- package used to plot graphs\n",
    "from matplotlib import pyplot as plt #<- another package used to plot graphs\n",
    "from itertools import cycle #<- package used to iterate down rows (used in step 5 to add tap column)\n",
    "import ipywidgets as widgets #<- widget tool to generate button and tab for graphs\n",
    "from IPython.display import display #<- displays widgets\n",
    "from ipyfilechooser import FileChooser\n",
    "# from tkinter import Tk, filedialog #<- Tkinter is a GUI package\n",
    "print(\"done step 1\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 2. Pick filepath (just run and click button from output)\n",
    "\n",
    "Run the following cell and click the button 'Select Folder' to pick a filepath.\n",
    "\n",
    "## Important: Later on, this script uses the total file path for each file to import and group data. That means if your folder has whatever your strain is named, the script will not work.\n",
    "\n",
    "(ex. if your folder has \"N2\" in it this script sees all files inside this folder as having the \"N2\" search key)\n",
    "\n",
    "## An easy fix is to just rename your folder to something else (make your strains lower-case, or just have the date)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "80c939e8fd09408f8fcef509593c0552",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "FileChooser(path='/Users/Joseph', filename='', title='', show_hidden=False, select_desc='Select', change_desc=â€¦"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "starting_directory = '/Users/Joseph'\n",
    "chooser = FileChooser(starting_directory)\n",
    "display(chooser)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/JosephDISK/Kristen-aging\n"
     ]
    }
   ],
   "source": [
    "print(chooser.selected_path)\n",
    "folder_path=chooser.selected_path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Obsolete\n",
    "# ### Select Folder App - After you run, you can select your folder for filepath\n",
    "# button = widgets.Button(description = 'Select Folder') #<- Creates a button variable\n",
    "# display(button) #<- displays this button on output\n",
    "# def select_folder(b): #<- This is an action. Requires a variable, so I put in an arbitrary one 'b'\n",
    "#     global folder_path #<- sets a variable as a global variable, not just within this action\n",
    "#     Tk().withdraw() #<- Tkinter likes to create annoying empty windows. This removes them\n",
    "#     folder_path = filedialog.askdirectory() #<- Opens up a file explorer window, and determines folder path\n",
    "#     Tk().update()#<- below\n",
    "#     Tk().destroy()#<- this and the line above it removes the file explorer window after a selection is made\n",
    "#     print(folder_path) #<- this helps confirm that this action was performed\n",
    "#     print('done step 2')\n",
    "# button.on_click(select_folder) #<- describes what happens when you click on this button"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 3. User Defined Variables (Add input here)\n",
    "\n",
    "Here, we add some constants to help you blaze through this code.\n",
    "\n",
    "3.1: Number of taps is pretty self-explanatory. How any taps does your experiment have? put in that number + 1 (N+1)!\n",
    "\n",
    "This may be a bit confusing, but this is due to some coding syntax that you don't have to worry about.\n",
    "\n",
    "3.2: Change your ISI number. This will be reflected in the name/title of the output figure."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[(1, (598, 602)), (2, (608, 612)), (3, (618, 622)), (4, (628, 632)), (5, (638, 642)), (6, (648, 652)), (7, (658, 662)), (8, (668, 672)), (9, (678, 682)), (10, (688, 692)), (11, (698, 702)), (12, (708, 712)), (13, (718, 722)), (14, (728, 732)), (15, (738, 742)), (16, (748, 752)), (17, (758, 762)), (18, (768, 772)), (19, (778, 782)), (20, (788, 792)), (21, (798, 802)), (22, (808, 812)), (23, (818, 822)), (24, (828, 832)), (25, (838, 842)), (26, (848, 852)), (27, (858, 862)), (28, (868, 872)), (29, (878, 882)), (30, (888, 892)), (31, (1188, 1191))]\n",
      "tap 1 tolerance=(598, 602)\n",
      "tap 2 tolerance=(608, 612)\n",
      "tap 3 tolerance=(618, 622)\n",
      "tap 4 tolerance=(628, 632)\n",
      "tap 5 tolerance=(638, 642)\n",
      "tap 6 tolerance=(648, 652)\n",
      "tap 7 tolerance=(658, 662)\n",
      "tap 8 tolerance=(668, 672)\n",
      "tap 9 tolerance=(678, 682)\n",
      "tap 10 tolerance=(688, 692)\n",
      "tap 11 tolerance=(698, 702)\n",
      "tap 12 tolerance=(708, 712)\n",
      "tap 13 tolerance=(718, 722)\n",
      "tap 14 tolerance=(728, 732)\n",
      "tap 15 tolerance=(738, 742)\n",
      "tap 16 tolerance=(748, 752)\n",
      "tap 17 tolerance=(758, 762)\n",
      "tap 18 tolerance=(768, 772)\n",
      "tap 19 tolerance=(778, 782)\n",
      "tap 20 tolerance=(788, 792)\n",
      "tap 21 tolerance=(798, 802)\n",
      "tap 22 tolerance=(808, 812)\n",
      "tap 23 tolerance=(818, 822)\n",
      "tap 24 tolerance=(828, 832)\n",
      "tap 25 tolerance=(838, 842)\n",
      "tap 26 tolerance=(848, 852)\n",
      "tap 27 tolerance=(858, 862)\n",
      "tap 28 tolerance=(868, 872)\n",
      "tap 29 tolerance=(878, 882)\n",
      "tap 30 tolerance=(888, 892)\n",
      "tap 31 tolerance=(1188, 1191)\n",
      "done step 3\n"
     ]
    }
   ],
   "source": [
    "\n",
    "number_of_taps = 30 ###<------ Taps in your experiment.\n",
    "\n",
    "number_taps = range(1, number_of_taps + 1)  #<- do not change this\n",
    "\n",
    "\n",
    "# if you have different ISIs in the same folder, then come back and change this \n",
    "# when you are graphing for the second set of data with the other ISI \n",
    "# (Generally data from same ISIs are graphed together)\n",
    "# If changing ISI mid-analysis, you can just skip straight to step 8 after running this cell again\n",
    "\n",
    "ISI = 10  ### <- What is your ISI? change accordingly\n",
    "first_tap = 600 ### <- when is your first tap? check your TRV files\n",
    "\n",
    "#Here, open up one of the trv files to determine the times for each of these taps. \n",
    "lower = np.arange(first_tap-2, first_tap-2+(number_of_taps*ISI), ISI) #(first tap, last tap+10s, ISI)\n",
    "upper = np.arange(first_tap+2, first_tap+2+(number_of_taps*ISI), ISI) #(first tap, last tap+10s, ISI)\n",
    "tolerances = list(zip(lower, upper)) \n",
    "taps = [i for i in range(1,number_of_taps+1)]\n",
    "\n",
    "#### Add 31st Tap Here --------------------------------------------------<------------\n",
    "tolerances.append((1188,1191))\n",
    "taps.append(31)\n",
    "\n",
    "\n",
    "#assign each tolerance to a tap number\n",
    "accurate_taps = list((zip(taps, tolerances)))\n",
    "print(accurate_taps)\n",
    "\n",
    "for pair in accurate_taps:\n",
    "   tap = pair[0]\n",
    "   tolerance = pair[1]\n",
    "   print(\"tap \"+str(tap), \"tolerance=\"+str(tolerance))\n",
    "\n",
    "print(\"done step 3\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 4. Constructing Filelist From Source File/Select File (Just run)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20240706_043802/N2_8x2_f96h20C_600s31x10s10s_A0707bb.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20240706_052432/N2_8x2_f96h20C_600s31x10s10s_A0706be.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20240706_112359/N2_8x2_f72h20C_600s31x10s10s_B0706ba.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20240706_121436/N2_8x2_f72h20C_600s31x10s10s_B0706bd.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20240720_054357/N2_7x2_f96h20C_600s31x10s10s_A0720bb.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20240720_060632/N2_7x2_f96h20C_600s31x10s10s_A0720bc.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20240720_065325/N2_7x2_f96h20C_600s31x10s10s_A0720be.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20240720_134220/N2_7x2_f72h20C_600s31x10s10s_B0720bd.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20240720_142615/N2_7x2_f72h20C_600s31x10s10s_B0720bf.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20241019_073624/N2_8x2_f96h20C_600s31x10s10s_A1019bc.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20241019_132858/N2_8x2_f72h20C_600s31x10s10s_B1019ba.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20241019_140124/N2_8X2_f96h20C_600s31x10s10s_C1019bb.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20241019_151008/N2_8x2_f72h20C_600s31x10s10s_B1019bd.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20241019_154120/N2_8X2_f96h20C_600s31x10s10s_C1019be.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20241102_055753/N2_8X2_f96h20C_600s31x10s10s_A1102cb.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20241102_073800/N2_8X2_f96h20C_600s31x10s10s_A1102ce.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20241102_121653/N2_8X2_f96h20C_600s31x10s10s_C1102ca.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20241102_133324/N2_8x2_f72h20C_600s31x10s10s_B1102cc.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20241102_140452/N2_8X2_f96h20C_600s31x10s10s_C1102cd.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20241109_024241/N2_8x2_f96h20C_600s31x10s10s_A1109bc.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20241109_083358/N2_8x2_f72h20C_600s31x10s10s_B1109ba.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20241109_090419/N2_8X2_f96h20C_600s31x10s10s_C1109bb.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20241109_101552/N2_8x2_f72h20C_600s31x10s10s_B1109bd.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20241109_104554/N2_8X2_f96h20C_600s31x10s10s_C1109be.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20240706_041149/N2_8x2_f96h20C_600s31x10s10s_A0707aa.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20240706_050228/N2_8x2_f96h20C_600s31x10s10s_A0706ad.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20240706_113054/N2_8x2_f96h20C_600s31x10s10s_C0707ab.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20240706_115010/N2_8x2_f72h20C_600s31x10s10s_B0706ac.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20240706_122136/N2_8x2_f96h20C_600s31x10s10s_C0706ae.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20240720_051711/N2_8x2_f96h20C_600s31x10s10s_A0720aa.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20240720_063033/N2_8x2_f96h20C_600s31x10s10s_A0720ad.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20240720_125456/N2_8x2_f72h20C_600s31x10s10s_B0720ab.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20240720_131858/N2_8x2_f72h20C_600s31x10s10s_B0720ac.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20240720_140510/N2_8x2_f72h20C_600s31x10s10s_B0720ae.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20241019_061931/N2_8x2_f96h20C_600s31x10s10s_A1019aa.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20241019_080041/N2_8x2_f96h20C_600s31x10s10s_A1019ad.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20241019_135620/N2_8x2_f72h20C_600s31x10s10s_B1019ab.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20241019_142509/N2_8X2_f96h20C_600s31x10s10s_C1019ac.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20241019_153613/N2_8x2_f72h20C_600s31x10s10s_B1019ae.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20241102_050250/N2_8X2_f96h20C_600s31x10s10s_A1102aa.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20241102_065039/N2_8X2_f96h20C_600s31x10s10s_A1102ad.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20241102_123946/N2_8x2_f72h20C_600s31x10s10s_B1102ab.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20241102_131202/N2_8X2_f96h20C_600s31x10s10s_C1102ac.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20241102_142409/N2_8x2_f72h20C_600s31x10s10s_B1102ae.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20241109_012502/N2_8x2_f96h20C_600s31x10s10s_A1109aa.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20241109_030652/N2_8x2_f96h20C_600s31x10s10s_A1109bad.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20241109_085923/N2_8x2_f72h20C_600s31x10s10s_B1109ab.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20241109_093046/N2_8X2_f96h20C_600s31x10s10s_C1109ac.trv', '/Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20241109_104055/N2_8x2_f72h20C_600s31x10s10s_B1109ae.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Apr2_2025/djr-1.1_tm918_7days/20250402_061631/FX00918_8x2_f96h20C_600s31x10s10s_A0402da.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Apr2_2025/djr-1.1_tm918_7days/20250402_074357/FX00918_8x2_f96h20C_600s31x10s10s_A0402dd.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Apr2_2025/djr-1.1_tm918_7days/20250402_141735/FX00918_8x2_f96h20C_600s31x10s10s_B0402db.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Apr2_2025/djr-1.1_tm918_7days/20250402_144759/FX00918_8x2_f96h20C_600s31x10s10s_C0402dc.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Apr2_2025/djr-1.1_tm918_96hrs/20250402_081614/FX00918_8x2_f96h20C_600s31x10s10s_A0402ce.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Apr2_2025/djr-1.1_tm918_96hrs/20250402_132617/FX00918_8x2_f96h20C_600s31x10s10s_C0402ca.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Apr2_2025/djr-1.1_tm918_96hrs/20250402_144425/FX00918_8x2_f96h20C_600s31x10s10s_B0402cc.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Apr2_2025/djr-1.1_tm918_96hrs/20250402_151210/FX00918_8x2_f96h20C_600s31x10s10s_C0402cd.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Apr2_2025/N2_N2_7days/20250402_065253/N2_8x2_f96h20C_600s31x10s10s_A0402bc.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Apr2_2025/N2_N2_7days/20250402_131402/N2_8x2_f96h20C_600s31x10s10s_B0402ba.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Apr2_2025/N2_N2_7days/20250402_135018/N2_8x2_f96h20C_600s31x10s10s_C0402bb.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Apr2_2025/N2_N2_7days/20250402_150836/N2_8x2_f96h20C_600s31x10s10s_B0402bd.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Apr2_2025/N2_N2_7days/20250402_154316/N2_8x2_f96h20C_600s31x10s10s_C0402be.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Apr2_2025/N2_N2_96hrs/20250402_054921/N2_8x2_f96h20C_600s31x10s10s_A0402aa.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Apr2_2025/N2_N2_96hrs/20250402_071938/N2_8x2_f96h20C_600s31x10s10s_A0402ad.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Apr2_2025/N2_N2_96hrs/20250402_134115/N2_8x2_f96h20C_600s31x10s10s_B0402ab.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Apr2_2025/N2_N2_96hrs/20250402_142107/N2_8x2_f96h20C_600s31x10s10s_C0402ac.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Apr2_2025/N2_N2_96hrs/20250402_154620/N2_8x2_f96h20C_600s31x10s10s_B0402ae.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/LRRK2_G2019S_7days/20250315_034050/WLZ3_8x2_f96h20C_600s31x10s10s_A0315dd.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/LRRK2_G2019S_7days/20250315_045908/WLZ3_8x2_f96h20C_600s31x10s10s_A0315da.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/LRRK2_G2019S_7days/20250315_110557/WLZ3_8x2_f72h20C_600s31x10s10s_B0315de.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/LRRK2_G2019S_7days/20250315_113356/WLZ3_8X2_f96h20C_600s31x10s10s_C0315dc.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/LRRK2_G2019S_7days/20250315_124913/WLZ3_8x2_f72h20C_600s31x10s10s_B0315db.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/LRRK2_G2019S_96hrs/20250315_043155/WLZ3_8x2_f96h20C_600s31x10s10s_A0315cb.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/LRRK2_G2019S_96hrs/20250315_061133/WLZ3_8x2_f96h20C_600s31x10s10s_A0315ce.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/LRRK2_G2019S_96hrs/20250315_110939/WLZ3_8X2_f96h20C_600s31x10s10s_C0315ca.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/LRRK2_G2019S_96hrs/20250315_122413/WLZ3_8x2_f72h20C_600s31x10s10s_B0315cc.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/LRRK2_G2019S_96hrs/20250315_125321/WLZ3_8X2_f96h20C_600s31x10s10s_C0315cd.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/N2_N2_7days/20250315_052358/N2_8x2_f96h20C_600s31x10s10s_A0315bc.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/N2_N2_7days/20250315_113014/N2_8x2_f72h20C_600s31x10s10s_B0315ba.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/N2_N2_7days/20250315_120044/N2_8X2_f96h20C_600s31x10s10s_C0315bb.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/N2_N2_7days/20250315_131403/N2_8x2_f72h20C_600s31x10s10s_B0315bd.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/N2_N2_7days/20250315_131752/N2_8X2_f96h20C_600s31x10s10s_C0315be.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/N2_N2_96hrs/20250315_040507/N2_8x2_f96h20C_600s31x10s10s_A0315aa.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/N2_N2_96hrs/20250315_054857/N2_8x2_f96h20C_600s31x10s10s_A0315ad.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/N2_N2_96hrs/20250315_115659/N2_8x2_f72h20C_600s31x10s10s_B0315ab.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/N2_N2_96hrs/20250315_122811/N2_8X2_f96h20C_600s31x10s10s_C0315ac.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/N2_N2_96hrs/20250315_133644/N2_8x2_f72h20C_600s31x10s10s_B0315ae.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/catp-7_tm4438_7days/20250319_023201/FX04438_8x2_f96h20C_600s31x10s10s_A0319da.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/catp-7_tm4438_7days/20250319_041819/FX04438_8x2_f96h20C_600s31x10s10s_A0319dd.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/catp-7_tm4438_7days/20250319_102217/FX04438_8x2_f96h20C_600s31x10s10s_B0319db.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/catp-7_tm4438_7days/20250319_105320/FX04438_8x2_f96h20C_600s31x10s10s_C0319dc.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/catp-7_tm4438_7days/20250319_120836/FX04438_8x2_f96h20C_600s31x10s10s_B0319de.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/catp-7_tm4438_96hrs/20250319_025716/FX04438_8x2_f96h20C_600s31x10s10s_A0319cb.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/catp-7_tm4438_96hrs/20250319_044336/FX04438_8x2_f96h20C_600s31x10s10s_A0319ce.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/catp-7_tm4438_96hrs/20250319_093548/FX04438_8x2_f96h20C_600s31x10s10s_C0319ca.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/catp-7_tm4438_96hrs/20250319_104940/FX04438_8x2_f96h20C_600s31x10s10s_B0319cc.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/catp-7_tm4438_96hrs/20250319_111741/FX04438_8x2_f96h20C_600s31x10s10s_C0319cd.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/N2_N2_7days/20250319_032429/N2_8x2_f96h20C_600s31x10s10s_A0319bc.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/N2_N2_7days/20250319_093206/N2_8x2_f96h20C_600s31x10s10s_B0319ba.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/N2_N2_7days/20250319_100044/N2_8x2_f96h20C_600s31x10s10s_C0319bb.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/N2_N2_7days/20250319_111404/N2_8x2_f96h20C_600s31x10s10s_B0319bd.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/N2_N2_7days/20250319_114705/N2_8x2_f96h20C_600s31x10s10s_C0319be.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/N2_N2_96hrs/20250319_020704/N2_8x2_f96h20C_600s31x10s10s_A0319aa.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/N2_N2_96hrs/20250319_034906/N2_8x2_f96h20C_600s31x10s10s_A0319ad.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/N2_N2_96hrs/20250319_095703/N2_8x2_f96h20C_600s31x10s10s_B0319ab.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/N2_N2_96hrs/20250319_102604/N2_8x2_f96h20C_600s31x10s10s_C0319ac.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/N2_N2_96hrs/20250319_114316/N2_8x2_f96h20C_600s31x10s10s_B0319ae.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov23_2024/N2_N2_7days/20241123_114101/N2_8x2_f72h20C_600s31x10s10s_B1123ba.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov23_2024/N2_N2_7days/20241123_121022/N2_8X2_f96h20C_600s31x10s10s_C1123bb.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov23_2024/N2_N2_7days/20241123_132531/N2_8x2_f72h20C_600s31x10s10s_B1123bd.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov23_2024/N2_N2_7days/20241123_135802/N2_8X2_f96h20C_600s31x10s10s_C1123be.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov23_2024/N2_N2_96hrs/20241123_043228/N2_8x2_f96h20C_600s31x10s10s_A1123aa.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov23_2024/N2_N2_96hrs/20241123_120531/N2_8x2_f72h20C_600s31x10s10s_B1123ab.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov23_2024/N2_N2_96hrs/20241123_123722/N2_8X2_f96h20C_600s31x10s10s_C1123ac.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov23_2024/N2_N2_96hrs/20241123_135313/N2_8x2_f72h20C_600s31x10s10s_B1123ae.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov23_2024/pdr-1_gk448_7days/20241123_052341/VC1024_8x2_f96h20C_600s31x10s10s_A1123cb.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov23_2024/pdr-1_gk448_7days/20241123_070956/VC1024_8x2_f96h20C_600s31x10s10s_A1123ce.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov23_2024/pdr-1_gk448_7days/20241123_114550/VC1024_8X2_f96h20C_600s31x10s10s_C1123ca.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov23_2024/pdr-1_gk448_7days/20241123_125854/VC1024_8x2_f72h20C_600s31x10s10s_B1123cc.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov23_2024/pdr-1_gk448_7days/20241123_133018/VC1024_8X2_f96h20C_600s31x10s10s_C1123cd.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov23_2024/pdr-1_gk448_96hrs/20241123_045654/VC1024_8x2_f96h20C_600s31x10s10s_A1123da.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov23_2024/pdr-1_gk448_96hrs/20241123_064437/VC1024_8x2_f96h20C_600s31x10s10s_A1123dd.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov23_2024/pdr-1_gk448_96hrs/20241123_123222/VC1024_8x2_f72h20C_600s31x10s10s_B1123db.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov23_2024/pdr-1_gk448_96hrs/20241123_141831/VC1024_8x2_f72h20C_600s31x10s10s_B1123de.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/lrk-1_tm1898_7days/20241102_053036/FX01898_8X2_f96h20C_600s31x10s10s_A1102da.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/lrk-1_tm1898_7days/20241102_071453/FX01898_8X2_f96h20C_600s31x10s10s_A1102dd.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/lrk-1_tm1898_7days/20241102_130656/FX01898_8x2_f72h20C_600s31x10s10s_B1102db.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/lrk-1_tm1898_7days/20241102_133821/FX01898_8X2_f96h20C_600s31x10s10s_C1102dc.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/lrk-1_tm1898_7days/20241102_144708/FX01898_8x2_f72h20C_600s31x10s10s_B1102de.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/lrk-1_tm1898_96hrs/20241102_062418/FX01898_8X2_f96h20C_600s31x10s10s_A1102bc.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/lrk-1_tm1898_96hrs/20241102_121159/FX01898_8x2_f72h20C_600s31x10s10s_B1102ba.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/lrk-1_tm1898_96hrs/20241102_124447/FX01898_8X2_f96h20C_600s31x10s10s_C1102bb.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/lrk-1_tm1898_96hrs/20241102_135948/FX01898_8x2_f72h20C_600s31x10s10s_B1102bd.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/lrk-1_tm1898_96hrs/20241102_142908/FX01898_8X2_f96h20C_600s31x10s10s_C1102be.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/N2_N2_7days/20241102_055753/N2_8X2_f96h20C_600s31x10s10s_A1102cb.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/N2_N2_7days/20241102_073800/N2_8X2_f96h20C_600s31x10s10s_A1102ce.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/N2_N2_7days/20241102_121653/N2_8X2_f96h20C_600s31x10s10s_C1102ca.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/N2_N2_7days/20241102_133324/N2_8x2_f72h20C_600s31x10s10s_B1102cc.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/N2_N2_7days/20241102_140452/N2_8X2_f96h20C_600s31x10s10s_C1102cd.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/N2_N2_96hrs/20241102_050250/N2_8X2_f96h20C_600s31x10s10s_A1102aa.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/N2_N2_96hrs/20241102_065039/N2_8X2_f96h20C_600s31x10s10s_A1102ad.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/N2_N2_96hrs/20241102_123946/N2_8x2_f72h20C_600s31x10s10s_B1102ab.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/N2_N2_96hrs/20241102_131202/N2_8X2_f96h20C_600s31x10s10s_C1102ac.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/N2_N2_96hrs/20241102_142409/N2_8x2_f72h20C_600s31x10s10s_B1102ae.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/N2_N2_7days/20241109_024241/N2_8x2_f96h20C_600s31x10s10s_A1109bc.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/N2_N2_7days/20241109_083358/N2_8x2_f72h20C_600s31x10s10s_B1109ba.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/N2_N2_7days/20241109_090419/N2_8X2_f96h20C_600s31x10s10s_C1109bb.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/N2_N2_7days/20241109_101552/N2_8x2_f72h20C_600s31x10s10s_B1109bd.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/N2_N2_7days/20241109_104554/N2_8X2_f96h20C_600s31x10s10s_C1109be.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/N2_N2_96hrs/20241109_012502/N2_8x2_f96h20C_600s31x10s10s_A1109aa.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/N2_N2_96hrs/20241109_030652/N2_8x2_f96h20C_600s31x10s10s_A1109bad.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/N2_N2_96hrs/20241109_085923/N2_8x2_f72h20C_600s31x10s10s_B1109ab.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/N2_N2_96hrs/20241109_093046/N2_8X2_f96h20C_600s31x10s10s_C1109ac.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/N2_N2_96hrs/20241109_104055/N2_8x2_f72h20C_600s31x10s10s_B1109ae.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/pink-1_tm1779_7days/20241109_015026/FX01779_10x2_f96h20C_600s31x10s10s_A1109da.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/pink-1_tm1779_7days/20241109_033200/FX01779_10x2_f96h20C_600s31x10s10s_A1109dd.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/pink-1_tm1779_7days/20241109_092547/FX01779_10x2_f72h20C_600s31x10s10s_B1109db.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/pink-1_tm1779_7days/20241109_095633/FX01779_10X2_f96h20C_600s31x10s10s_C1109dc.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/pink-1_tm1779_7days/20241109_110412/FX01779_10x2_f72h20C_600s31x10s10s_B1109de.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/pink-1_tm1779_96hrs/20241109_021642/FX01779_10x2_f96h20C_600s31x10s10s_A1109cb.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/pink-1_tm1779_96hrs/20241109_035518/FX01779_10x2_f96h20C_600s31x10s10s_A1109ce.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/pink-1_tm1779_96hrs/20241109_083854/FX01779_10X2_f96h20C_600s31x10s10s_C1109ca.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/pink-1_tm1779_96hrs/20241109_095136/FX01779_10x2_f72h20C_600s31x10s10s_B1109cc.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/pink-1_tm1779_96hrs/20241109_102106/FX01779_10X2_f96h20C_600s31x10s10s_C1109cd.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/N2_N2_7days/20241019_073624/N2_8x2_f96h20C_600s31x10s10s_A1019bc.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/N2_N2_7days/20241019_132858/N2_8x2_f72h20C_600s31x10s10s_B1019ba.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/N2_N2_7days/20241019_140124/N2_8X2_f96h20C_600s31x10s10s_C1019bb.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/N2_N2_7days/20241019_151008/N2_8x2_f72h20C_600s31x10s10s_B1019bd.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/N2_N2_7days/20241019_154120/N2_8X2_f96h20C_600s31x10s10s_C1019be.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/N2_N2_96hrs/20241019_061931/N2_8x2_f96h20C_600s31x10s10s_A1019aa.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/N2_N2_96hrs/20241019_080041/N2_8x2_f96h20C_600s31x10s10s_A1019ad.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/N2_N2_96hrs/20241019_135620/N2_8x2_f72h20C_600s31x10s10s_B1019ab.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/N2_N2_96hrs/20241019_142509/N2_8X2_f96h20C_600s31x10s10s_C1019ac.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/N2_N2_96hrs/20241019_153613/N2_8x2_f72h20C_600s31x10s10s_B1019ae.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/pink-1_ok3538_7days/20241019_064642/RB2547_11x2_f96h20C_600s31x10s10s_A1019da.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/pink-1_ok3538_7days/20241019_082644/RB2547_11x2_f96h20C_600s31x10s10s_A1019dd.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/pink-1_ok3538_7days/20241019_141958/RB2547_11x2_f72h20C_600s31x10s10s_B1019db.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/pink-1_ok3538_7days/20241019_145104/RB2547_11X2_f96h20C_600s31x10s10s_C1019dc.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/pink-1_ok3538_7days/20241019_160023/RB2547_11x2_f72h20C_600s31x10s10s_B1019de.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/pink-1_ok3538_96hrs/20241019_071026/RB2547_10x2_f96h20C_600s31x10s10s_A1019cb.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/pink-1_ok3538_96hrs/20241019_085134/RB2547_10x2_f96h20C_600s31x10s10s_A1019ce.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/pink-1_ok3538_96hrs/20241019_133403/RB2547_11X2_f96h20C_600s31x10s10s_C1019ca.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/pink-1_ok3538_96hrs/20241019_144556/RB2547_10x2_f72h20C_600s31x10s10s_B1019cc.trv', '/Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/pink-1_ok3538_96hrs/20241019_151514/RB2547_10X2_f96h20C_600s31x10s10s_C1019cd.trv']\n",
      "done step 4\n"
     ]
    }
   ],
   "source": [
    "#folder_path = '/Users/Joseph/Desktop/AVR14_10sISI' #- manual folder path if Tkinter is acting up\n",
    "\n",
    "os.chdir(folder_path) #<- setting your working directory so that your images will be saved here\n",
    "\n",
    "filelist = list() #<- empty list\n",
    "for root, dirs, files in os.walk(folder_path): #<- this for loop goes through your folder \n",
    "    for name in files:\n",
    "        if name.endswith('.trv'): #<- and takes out all files with a .trv (file that contains your data)\n",
    "            filepath = os.path.join(root, name) #<- Notes down the file path of each data file\n",
    "            filelist.append(filepath) #<- saves it into the list\n",
    "    \n",
    "print(filelist)\n",
    "print('done step 4')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# string = '/Volumes/JOSEPH/PD_Screen/bas-1_tm351/20220809_103216/LC33_10x2_f72h20C_600s31x10s10s_B0809db.trv'\n",
    "# print(string.split('/')[-3])\n",
    "# print(string.split('/')[-2].split('_')[0])\n",
    "# print(string.split('/')[-1].split('_')[-1].split('.')[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['LRRK2_G2019S_7days' 'LRRK2_G2019S_96hrs' 'N2_N2_7days' 'N2_N2_96hrs'\n",
      " 'catp-7_tm4438_7days' 'catp-7_tm4438_96hrs' 'djr-1.1_tm918_7days'\n",
      " 'djr-1.1_tm918_96hrs' 'lrk-1_tm1898_7days' 'lrk-1_tm1898_96hrs'\n",
      " 'pdr-1_gk448_7days' 'pdr-1_gk448_96hrs' 'pink-1_ok3538_7days'\n",
      " 'pink-1_ok3538_96hrs' 'pink-1_tm1779_7days' 'pink-1_tm1779_96hrs']\n",
      "16\n",
      "[(1, 'LRRK2_G2019S_7days'), (2, 'LRRK2_G2019S_96hrs'), (3, 'N2_N2_7days'), (4, 'N2_N2_96hrs'), (5, 'catp-7_tm4438_7days'), (6, 'catp-7_tm4438_96hrs'), (7, 'djr-1.1_tm918_7days'), (8, 'djr-1.1_tm918_96hrs'), (9, 'lrk-1_tm1898_7days'), (10, 'lrk-1_tm1898_96hrs'), (11, 'pdr-1_gk448_7days'), (12, 'pdr-1_gk448_96hrs'), (13, 'pink-1_ok3538_7days'), (14, 'pink-1_ok3538_96hrs'), (15, 'pink-1_tm1779_7days'), (16, 'pink-1_tm1779_96hrs')]\n",
      "{1: 'LRRK2_G2019S_7days', 2: 'LRRK2_G2019S_96hrs', 3: 'N2_N2_7days', 4: 'N2_N2_96hrs', 5: 'catp-7_tm4438_7days', 6: 'catp-7_tm4438_96hrs', 7: 'djr-1.1_tm918_7days', 8: 'djr-1.1_tm918_96hrs', 9: 'lrk-1_tm1898_7days', 10: 'lrk-1_tm1898_96hrs', 11: 'pdr-1_gk448_7days', 12: 'pdr-1_gk448_96hrs', 13: 'pink-1_ok3538_7days', 14: 'pink-1_ok3538_96hrs', 15: 'pink-1_tm1779_7days', 16: 'pink-1_tm1779_96hrs'}\n"
     ]
    }
   ],
   "source": [
    "genotype=[]\n",
    "for f in filelist:\n",
    "    genotype.append(f.split('/')[-3])\n",
    "# print(np.unique(genotype))\n",
    "genotypes=np.unique(genotype)\n",
    "print(genotypes)\n",
    "print(len(np.unique(genotypes)))\n",
    "strainNs = enumerate(genotypes,1)\n",
    "strain_enum=list(strainNs)\n",
    "print(strain_enum)\n",
    "StrainNames=dict(strain_enum)\n",
    "print(StrainNames)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 5. Process Data Function (Just Run)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "done step 5\n"
     ]
    }
   ],
   "source": [
    "def ProcessData(strain): #<- an example of a user-defined function\n",
    "    strain_filelist = [x for x in filelist if strain in x] #<- goes through the list and filters for keyword\n",
    "    Strain_N = len(strain_filelist) #<- N per strain, or number of plates\n",
    "    Plate_N = 1\n",
    "    print(f'this strain/treatment has {Strain_N} plates') #<- will output as the first number\n",
    "    if Strain_N == 0:\n",
    "        raise AssertionError ('{} is not a good identifier'.format(strain))\n",
    "    else:\n",
    "        pass\n",
    "        print(f'now working on strain {strain}')\n",
    "        strain_filelist = [file for file in filelist if strain in file]\n",
    "        df_list=[]\n",
    "        for file in strain_filelist:\n",
    "            if file.split('/')[-1].startswith('._'):\n",
    "                pass\n",
    "            else:\n",
    "                print(f\"working on file {file}\")\n",
    "                df= pd.read_csv(file, sep=' ', header = None, encoding_errors='ignore')\n",
    "                df['Plate_id'] = file.split('/')[-1].split('_')[-1].split('.')[0]\n",
    "                df['Date'] = file.split('/')[-2].split('_')[0]\n",
    "                df['Screen'] = file.split('/')[-4]\n",
    "                df_list.append(df)\n",
    "        DF_Total = pd.concat(df_list, ignore_index = True)\n",
    "#     for f in strain_filelist:\n",
    "#         DF_Total = pd.concat(pd.read_csv(f, sep=' ', skiprows = 4, header = None))\n",
    "        # DF_Total = pd.concat([pd.read_csv(f, sep=' ', header = None, encoding_errors='ignore') for f in strain_filelist],\n",
    "        #               ignore_index=True) #<- imports your data files\n",
    "#         DF_Total = DF_Total.dropna(axis = 1) #<- cleans your data\n",
    "        DF_Total = DF_Total.rename( #<- more cleaning\n",
    "                    {0:'time',\n",
    "                    2:'rev_before',\n",
    "                    3:'no_rev',\n",
    "                    4:'stim_rev',\n",
    "                    7:'dist',\n",
    "                    8:'dist_std',\n",
    "                    9:'dist_stderr',\n",
    "                    11:'dist_0th',\n",
    "                    12:'dist_1st',\n",
    "                    13:'dist_2nd',\n",
    "                    14:'dist_3rd',\n",
    "                    15:'dist_100th',\n",
    "                    18:'dura',\n",
    "                    19:'dura_std',\n",
    "                    20:'dura_stderr',\n",
    "                    22:'dura_0th',\n",
    "                    23:'dura_1st',\n",
    "                    24:'dura_2nd',\n",
    "                    25:'dura_3rd',\n",
    "                    26:'dura_100th'}, axis=1)\n",
    "        #check function here for NaN Columns\n",
    "        DF_Total['plate'] = 0\n",
    "        DF_Total['prob'] = DF_Total['stim_rev']/ (DF_Total['no_rev'] + DF_Total['stim_rev']) #<- calculate prob\n",
    "        DF_Total['speed'] = DF_Total['dist']/DF_Total['dura'] #<- calculate speed\n",
    "        DF_Total_rows = int(DF_Total.shape[0])\n",
    "        print(f'this strain/treatment has {DF_Total_rows} total taps') #<- Outputs as the second number. Check if you are missing taps!\n",
    "        DF_Final = DF_Total[[\"time\", \"dura\", \"dist\", \"prob\", \"speed\", \"plate\", \"Date\",\"Plate_id\",\"Screen\"]].copy()\n",
    "\n",
    "    return{\n",
    "            'N': Strain_N,\n",
    "            'Confirm':DF_Total,\n",
    "            'Final': DF_Final}\n",
    "    print('done!')\n",
    "\n",
    "\n",
    "\n",
    "def assign_taps(DF, tolerances):\n",
    "    DF['taps'] = np.nan\n",
    "    for taps, tolerance in enumerate(tolerances): #[(99, 101), (109,111), ...]\n",
    "        tap_lower,tap_upper = tolerance\n",
    "        TimesInTapRange = DF['time'].between(tap_lower,tap_upper, inclusive=\"both\")\n",
    "        DF.loc[TimesInTapRange,'taps'] = taps+1 #set the tap to i where times are between\n",
    "\n",
    "\n",
    "\n",
    "    \n",
    "def insert_plates(df):   \n",
    "    \"\"\"This function inserts a plate column into a dataframe.\n",
    "    \n",
    "    :param df: any dataframe\n",
    "    :type: pandas.core.frame.DataFrame\n",
    "    \n",
    "    :return: dataframe with a plate column\n",
    "    :type: pandas.core.frame.DataFrame\n",
    "    \"\"\"\n",
    "    df['plate']=(df['taps'] ==1).cumsum()\n",
    "\n",
    "\n",
    "            \n",
    "print('done step 5')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 6.1 Process Data (PLEASE READ, Add input here)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Here is the hardest part - From your naming convention, you need to pick a unique identifier for each group.\n",
    "\n",
    "This means that all of names of your files for that strain should have that in common but is not commone with across all other files! If you did a good job naming your files and following a good naming convention, this should be easy.\n",
    "\n",
    "## Be careful and really look hard in your naming structure. Note you want an unique identifier in the entire file path for the same group of files. An easy mistake is to have the strain name in the overall folder name, in this case if you use your strain name as a keyword it would include all files in that folder!\n",
    "\n",
    "For example, if all your N2 files have a certain pattern like \"N2_5x4\" in this following example:\n",
    "'/Users/Joseph/Desktop/AVR14_10sISI_TapHab_0710_2019/N2/20190710_141740/N2_5x4_f94h20c_100s30x10s10s_C0710ab.trv'\n",
    "\n",
    "Then you need to set that identifier for the strain keyword:\n",
    "'Strain_1' = 'N2_5x4'\n",
    "\n",
    "## Depending on how many strains you are running for comparison, you may need to add/delete some lines!\n",
    "\n",
    "## You are not naming your data groups here, we have a step for that later!\n",
    "## Here, you want to note down ALL the strains you have in the folder\n",
    "\n",
    "If you have just 2 strains, add hashtags (#) in front of the lines you do not need.\n",
    "If you need more strains, just add more lines, following the same format!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# #Format: 'Strain_#'' = 'unique_identifier'\n",
    "\n",
    "# ### Make into dictionary\n",
    "# StrainNames = {\n",
    "#     'Strain_1' : 'N2',  #<- each strain will be designated to a unique identifier here\n",
    "#     'Strain_2' : 'cat-2_e1112',\n",
    "#     'Strain_3' : 'cat-2_n4547',\n",
    "#     'Strain_4' : 'cat-2_tm2261',\n",
    "#     'Strain_5' : 'dat-1_ok157',\n",
    "#     'Strain_6' : 'cat-1_ok411',\n",
    "#     'Strain_7' : 'pdr-1_gk448', #<- empty entries are for those hardcore trackers that tracks this many strains\n",
    "#     'Strain_8' : 'pdr-1_tm395',\n",
    "#     'Strain_9' : 'pdr-1_tm598',\n",
    "#     'Strain_10' : 'dop-1_vs101',\n",
    "#     'Strain_11' : 'dop-3_ok295',\n",
    "#     'Strain_12' : 'snb-5_ok1434',\n",
    "#     'Strain_13' : 'unc-10_ad591',\n",
    "# #     'Strain_14' : '',\n",
    "# #     'Strain_15' : '',\n",
    "# }\n",
    "# #...etc, etc\n",
    "\n",
    "# print('done step 6.1')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['LRRK2_G2019S_7days', 'LRRK2_G2019S_96hrs', 'N2_N2_7days', 'N2_N2_96hrs', 'catp-7_tm4438_7days', 'catp-7_tm4438_96hrs', 'djr-1.1_tm918_7days', 'djr-1.1_tm918_96hrs', 'lrk-1_tm1898_7days', 'lrk-1_tm1898_96hrs', 'pdr-1_gk448_7days', 'pdr-1_gk448_96hrs', 'pink-1_ok3538_7days', 'pink-1_ok3538_96hrs', 'pink-1_tm1779_7days', 'pink-1_tm1779_96hrs']\n",
      "<class 'list'>\n",
      "<class 'numpy.str_'>\n"
     ]
    }
   ],
   "source": [
    "print(list(StrainNames.values()))\n",
    "print(type(list(StrainNames.values())))\n",
    "print(type(list(StrainNames.values())[0]))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 6.2 Process Data (just run this cell)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# print(ProcessData('F53B2.5_ok226')['Final'])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# with open('/Volumes/JOSEPH/PD_Screen/F53B2.5_ok226/20220524_141642/ZE1_10x2_f72h20C_600s31x10s10s_B0520ba.trv') as f:\n",
    "#     print(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# import threading"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "this strain/treatment has 5 plates\n",
      "now working on strain LRRK2_G2019S_7days\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/LRRK2_G2019S_7days/20250315_034050/WLZ3_8x2_f96h20C_600s31x10s10s_A0315dd.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/LRRK2_G2019S_7days/20250315_045908/WLZ3_8x2_f96h20C_600s31x10s10s_A0315da.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/LRRK2_G2019S_7days/20250315_110557/WLZ3_8x2_f72h20C_600s31x10s10s_B0315de.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/LRRK2_G2019S_7days/20250315_113356/WLZ3_8X2_f96h20C_600s31x10s10s_C0315dc.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/LRRK2_G2019S_7days/20250315_124913/WLZ3_8x2_f72h20C_600s31x10s10s_B0315db.trv\n",
      "this strain/treatment has 155 total taps\n",
      "this strain/treatment has 5 plates\n",
      "now working on strain LRRK2_G2019S_96hrs\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/LRRK2_G2019S_96hrs/20250315_043155/WLZ3_8x2_f96h20C_600s31x10s10s_A0315cb.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/LRRK2_G2019S_96hrs/20250315_061133/WLZ3_8x2_f96h20C_600s31x10s10s_A0315ce.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/LRRK2_G2019S_96hrs/20250315_110939/WLZ3_8X2_f96h20C_600s31x10s10s_C0315ca.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/LRRK2_G2019S_96hrs/20250315_122413/WLZ3_8x2_f72h20C_600s31x10s10s_B0315cc.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/LRRK2_G2019S_96hrs/20250315_125321/WLZ3_8X2_f96h20C_600s31x10s10s_C0315cd.trv\n",
      "this strain/treatment has 154 total taps\n",
      "this strain/treatment has 58 plates\n",
      "now working on strain N2_N2_7days\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20240706_043802/N2_8x2_f96h20C_600s31x10s10s_A0707bb.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20240706_052432/N2_8x2_f96h20C_600s31x10s10s_A0706be.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20240706_112359/N2_8x2_f72h20C_600s31x10s10s_B0706ba.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20240706_121436/N2_8x2_f72h20C_600s31x10s10s_B0706bd.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20240720_054357/N2_7x2_f96h20C_600s31x10s10s_A0720bb.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20240720_060632/N2_7x2_f96h20C_600s31x10s10s_A0720bc.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20240720_065325/N2_7x2_f96h20C_600s31x10s10s_A0720be.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20240720_134220/N2_7x2_f72h20C_600s31x10s10s_B0720bd.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20240720_142615/N2_7x2_f72h20C_600s31x10s10s_B0720bf.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20241019_073624/N2_8x2_f96h20C_600s31x10s10s_A1019bc.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20241019_132858/N2_8x2_f72h20C_600s31x10s10s_B1019ba.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20241019_140124/N2_8X2_f96h20C_600s31x10s10s_C1019bb.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20241019_151008/N2_8x2_f72h20C_600s31x10s10s_B1019bd.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20241019_154120/N2_8X2_f96h20C_600s31x10s10s_C1019be.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20241102_055753/N2_8X2_f96h20C_600s31x10s10s_A1102cb.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20241102_073800/N2_8X2_f96h20C_600s31x10s10s_A1102ce.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20241102_121653/N2_8X2_f96h20C_600s31x10s10s_C1102ca.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20241102_133324/N2_8x2_f72h20C_600s31x10s10s_B1102cc.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20241102_140452/N2_8X2_f96h20C_600s31x10s10s_C1102cd.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20241109_024241/N2_8x2_f96h20C_600s31x10s10s_A1109bc.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20241109_083358/N2_8x2_f72h20C_600s31x10s10s_B1109ba.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20241109_090419/N2_8X2_f96h20C_600s31x10s10s_C1109bb.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20241109_101552/N2_8x2_f72h20C_600s31x10s10s_B1109bd.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_7days/20241109_104554/N2_8X2_f96h20C_600s31x10s10s_C1109be.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Apr2_2025/N2_N2_7days/20250402_065253/N2_8x2_f96h20C_600s31x10s10s_A0402bc.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Apr2_2025/N2_N2_7days/20250402_131402/N2_8x2_f96h20C_600s31x10s10s_B0402ba.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Apr2_2025/N2_N2_7days/20250402_135018/N2_8x2_f96h20C_600s31x10s10s_C0402bb.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Apr2_2025/N2_N2_7days/20250402_150836/N2_8x2_f96h20C_600s31x10s10s_B0402bd.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Apr2_2025/N2_N2_7days/20250402_154316/N2_8x2_f96h20C_600s31x10s10s_C0402be.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/N2_N2_7days/20250315_052358/N2_8x2_f96h20C_600s31x10s10s_A0315bc.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/N2_N2_7days/20250315_113014/N2_8x2_f72h20C_600s31x10s10s_B0315ba.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/N2_N2_7days/20250315_120044/N2_8X2_f96h20C_600s31x10s10s_C0315bb.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/N2_N2_7days/20250315_131403/N2_8x2_f72h20C_600s31x10s10s_B0315bd.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/N2_N2_7days/20250315_131752/N2_8X2_f96h20C_600s31x10s10s_C0315be.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/N2_N2_7days/20250319_032429/N2_8x2_f96h20C_600s31x10s10s_A0319bc.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/N2_N2_7days/20250319_093206/N2_8x2_f96h20C_600s31x10s10s_B0319ba.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/N2_N2_7days/20250319_100044/N2_8x2_f96h20C_600s31x10s10s_C0319bb.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/N2_N2_7days/20250319_111404/N2_8x2_f96h20C_600s31x10s10s_B0319bd.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/N2_N2_7days/20250319_114705/N2_8x2_f96h20C_600s31x10s10s_C0319be.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov23_2024/N2_N2_7days/20241123_114101/N2_8x2_f72h20C_600s31x10s10s_B1123ba.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov23_2024/N2_N2_7days/20241123_121022/N2_8X2_f96h20C_600s31x10s10s_C1123bb.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov23_2024/N2_N2_7days/20241123_132531/N2_8x2_f72h20C_600s31x10s10s_B1123bd.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov23_2024/N2_N2_7days/20241123_135802/N2_8X2_f96h20C_600s31x10s10s_C1123be.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/N2_N2_7days/20241102_055753/N2_8X2_f96h20C_600s31x10s10s_A1102cb.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/N2_N2_7days/20241102_073800/N2_8X2_f96h20C_600s31x10s10s_A1102ce.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/N2_N2_7days/20241102_121653/N2_8X2_f96h20C_600s31x10s10s_C1102ca.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/N2_N2_7days/20241102_133324/N2_8x2_f72h20C_600s31x10s10s_B1102cc.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/N2_N2_7days/20241102_140452/N2_8X2_f96h20C_600s31x10s10s_C1102cd.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/N2_N2_7days/20241109_024241/N2_8x2_f96h20C_600s31x10s10s_A1109bc.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/N2_N2_7days/20241109_083358/N2_8x2_f72h20C_600s31x10s10s_B1109ba.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/N2_N2_7days/20241109_090419/N2_8X2_f96h20C_600s31x10s10s_C1109bb.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/N2_N2_7days/20241109_101552/N2_8x2_f72h20C_600s31x10s10s_B1109bd.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/N2_N2_7days/20241109_104554/N2_8X2_f96h20C_600s31x10s10s_C1109be.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/N2_N2_7days/20241019_073624/N2_8x2_f96h20C_600s31x10s10s_A1019bc.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/N2_N2_7days/20241019_132858/N2_8x2_f72h20C_600s31x10s10s_B1019ba.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/N2_N2_7days/20241019_140124/N2_8X2_f96h20C_600s31x10s10s_C1019bb.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/N2_N2_7days/20241019_151008/N2_8x2_f72h20C_600s31x10s10s_B1019bd.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/N2_N2_7days/20241019_154120/N2_8X2_f96h20C_600s31x10s10s_C1019be.trv\n",
      "this strain/treatment has 1796 total taps\n",
      "this strain/treatment has 59 plates\n",
      "now working on strain N2_N2_96hrs\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20240706_041149/N2_8x2_f96h20C_600s31x10s10s_A0707aa.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20240706_050228/N2_8x2_f96h20C_600s31x10s10s_A0706ad.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20240706_113054/N2_8x2_f96h20C_600s31x10s10s_C0707ab.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20240706_115010/N2_8x2_f72h20C_600s31x10s10s_B0706ac.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20240706_122136/N2_8x2_f96h20C_600s31x10s10s_C0706ae.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20240720_051711/N2_8x2_f96h20C_600s31x10s10s_A0720aa.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20240720_063033/N2_8x2_f96h20C_600s31x10s10s_A0720ad.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20240720_125456/N2_8x2_f72h20C_600s31x10s10s_B0720ab.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20240720_131858/N2_8x2_f72h20C_600s31x10s10s_B0720ac.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20240720_140510/N2_8x2_f72h20C_600s31x10s10s_B0720ae.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20241019_061931/N2_8x2_f96h20C_600s31x10s10s_A1019aa.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20241019_080041/N2_8x2_f96h20C_600s31x10s10s_A1019ad.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20241019_135620/N2_8x2_f72h20C_600s31x10s10s_B1019ab.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20241019_142509/N2_8X2_f96h20C_600s31x10s10s_C1019ac.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20241019_153613/N2_8x2_f72h20C_600s31x10s10s_B1019ae.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20241102_050250/N2_8X2_f96h20C_600s31x10s10s_A1102aa.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20241102_065039/N2_8X2_f96h20C_600s31x10s10s_A1102ad.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20241102_123946/N2_8x2_f72h20C_600s31x10s10s_B1102ab.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20241102_131202/N2_8X2_f96h20C_600s31x10s10s_C1102ac.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20241102_142409/N2_8x2_f72h20C_600s31x10s10s_B1102ae.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20241109_012502/N2_8x2_f96h20C_600s31x10s10s_A1109aa.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20241109_030652/N2_8x2_f96h20C_600s31x10s10s_A1109bad.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20241109_085923/N2_8x2_f72h20C_600s31x10s10s_B1109ab.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20241109_093046/N2_8X2_f96h20C_600s31x10s10s_C1109ac.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/Kristen-N2/N2_N2_96hrs/20241109_104055/N2_8x2_f72h20C_600s31x10s10s_B1109ae.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Apr2_2025/N2_N2_96hrs/20250402_054921/N2_8x2_f96h20C_600s31x10s10s_A0402aa.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Apr2_2025/N2_N2_96hrs/20250402_071938/N2_8x2_f96h20C_600s31x10s10s_A0402ad.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Apr2_2025/N2_N2_96hrs/20250402_134115/N2_8x2_f96h20C_600s31x10s10s_B0402ab.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Apr2_2025/N2_N2_96hrs/20250402_142107/N2_8x2_f96h20C_600s31x10s10s_C0402ac.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Apr2_2025/N2_N2_96hrs/20250402_154620/N2_8x2_f96h20C_600s31x10s10s_B0402ae.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/N2_N2_96hrs/20250315_040507/N2_8x2_f96h20C_600s31x10s10s_A0315aa.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/N2_N2_96hrs/20250315_054857/N2_8x2_f96h20C_600s31x10s10s_A0315ad.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/N2_N2_96hrs/20250315_115659/N2_8x2_f72h20C_600s31x10s10s_B0315ab.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/N2_N2_96hrs/20250315_122811/N2_8X2_f96h20C_600s31x10s10s_C0315ac.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar15_2025/N2_N2_96hrs/20250315_133644/N2_8x2_f72h20C_600s31x10s10s_B0315ae.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/N2_N2_96hrs/20250319_020704/N2_8x2_f96h20C_600s31x10s10s_A0319aa.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/N2_N2_96hrs/20250319_034906/N2_8x2_f96h20C_600s31x10s10s_A0319ad.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/N2_N2_96hrs/20250319_095703/N2_8x2_f96h20C_600s31x10s10s_B0319ab.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/N2_N2_96hrs/20250319_102604/N2_8x2_f96h20C_600s31x10s10s_C0319ac.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/N2_N2_96hrs/20250319_114316/N2_8x2_f96h20C_600s31x10s10s_B0319ae.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov23_2024/N2_N2_96hrs/20241123_043228/N2_8x2_f96h20C_600s31x10s10s_A1123aa.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov23_2024/N2_N2_96hrs/20241123_120531/N2_8x2_f72h20C_600s31x10s10s_B1123ab.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov23_2024/N2_N2_96hrs/20241123_123722/N2_8X2_f96h20C_600s31x10s10s_C1123ac.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov23_2024/N2_N2_96hrs/20241123_135313/N2_8x2_f72h20C_600s31x10s10s_B1123ae.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/N2_N2_96hrs/20241102_050250/N2_8X2_f96h20C_600s31x10s10s_A1102aa.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/N2_N2_96hrs/20241102_065039/N2_8X2_f96h20C_600s31x10s10s_A1102ad.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/N2_N2_96hrs/20241102_123946/N2_8x2_f72h20C_600s31x10s10s_B1102ab.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/N2_N2_96hrs/20241102_131202/N2_8X2_f96h20C_600s31x10s10s_C1102ac.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/N2_N2_96hrs/20241102_142409/N2_8x2_f72h20C_600s31x10s10s_B1102ae.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/N2_N2_96hrs/20241109_012502/N2_8x2_f96h20C_600s31x10s10s_A1109aa.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/N2_N2_96hrs/20241109_030652/N2_8x2_f96h20C_600s31x10s10s_A1109bad.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/N2_N2_96hrs/20241109_085923/N2_8x2_f72h20C_600s31x10s10s_B1109ab.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/N2_N2_96hrs/20241109_093046/N2_8X2_f96h20C_600s31x10s10s_C1109ac.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/N2_N2_96hrs/20241109_104055/N2_8x2_f72h20C_600s31x10s10s_B1109ae.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/N2_N2_96hrs/20241019_061931/N2_8x2_f96h20C_600s31x10s10s_A1019aa.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/N2_N2_96hrs/20241019_080041/N2_8x2_f96h20C_600s31x10s10s_A1019ad.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/N2_N2_96hrs/20241019_135620/N2_8x2_f72h20C_600s31x10s10s_B1019ab.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/N2_N2_96hrs/20241019_142509/N2_8X2_f96h20C_600s31x10s10s_C1019ac.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/N2_N2_96hrs/20241019_153613/N2_8x2_f72h20C_600s31x10s10s_B1019ae.trv\n",
      "this strain/treatment has 1825 total taps\n",
      "this strain/treatment has 5 plates\n",
      "now working on strain catp-7_tm4438_7days\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/catp-7_tm4438_7days/20250319_023201/FX04438_8x2_f96h20C_600s31x10s10s_A0319da.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/catp-7_tm4438_7days/20250319_041819/FX04438_8x2_f96h20C_600s31x10s10s_A0319dd.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/catp-7_tm4438_7days/20250319_102217/FX04438_8x2_f96h20C_600s31x10s10s_B0319db.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/catp-7_tm4438_7days/20250319_105320/FX04438_8x2_f96h20C_600s31x10s10s_C0319dc.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/catp-7_tm4438_7days/20250319_120836/FX04438_8x2_f96h20C_600s31x10s10s_B0319de.trv\n",
      "this strain/treatment has 154 total taps\n",
      "this strain/treatment has 5 plates\n",
      "now working on strain catp-7_tm4438_96hrs\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/catp-7_tm4438_96hrs/20250319_025716/FX04438_8x2_f96h20C_600s31x10s10s_A0319cb.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/catp-7_tm4438_96hrs/20250319_044336/FX04438_8x2_f96h20C_600s31x10s10s_A0319ce.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/catp-7_tm4438_96hrs/20250319_093548/FX04438_8x2_f96h20C_600s31x10s10s_C0319ca.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/catp-7_tm4438_96hrs/20250319_104940/FX04438_8x2_f96h20C_600s31x10s10s_B0319cc.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Mar19_2025/catp-7_tm4438_96hrs/20250319_111741/FX04438_8x2_f96h20C_600s31x10s10s_C0319cd.trv\n",
      "this strain/treatment has 155 total taps\n",
      "this strain/treatment has 4 plates\n",
      "now working on strain djr-1.1_tm918_7days\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Apr2_2025/djr-1.1_tm918_7days/20250402_061631/FX00918_8x2_f96h20C_600s31x10s10s_A0402da.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Apr2_2025/djr-1.1_tm918_7days/20250402_074357/FX00918_8x2_f96h20C_600s31x10s10s_A0402dd.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Apr2_2025/djr-1.1_tm918_7days/20250402_141735/FX00918_8x2_f96h20C_600s31x10s10s_B0402db.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Apr2_2025/djr-1.1_tm918_7days/20250402_144759/FX00918_8x2_f96h20C_600s31x10s10s_C0402dc.trv\n",
      "this strain/treatment has 124 total taps\n",
      "this strain/treatment has 4 plates\n",
      "now working on strain djr-1.1_tm918_96hrs\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Apr2_2025/djr-1.1_tm918_96hrs/20250402_081614/FX00918_8x2_f96h20C_600s31x10s10s_A0402ce.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Apr2_2025/djr-1.1_tm918_96hrs/20250402_132617/FX00918_8x2_f96h20C_600s31x10s10s_C0402ca.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Apr2_2025/djr-1.1_tm918_96hrs/20250402_144425/FX00918_8x2_f96h20C_600s31x10s10s_B0402cc.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Apr2_2025/djr-1.1_tm918_96hrs/20250402_151210/FX00918_8x2_f96h20C_600s31x10s10s_C0402cd.trv\n",
      "this strain/treatment has 124 total taps\n",
      "this strain/treatment has 5 plates\n",
      "now working on strain lrk-1_tm1898_7days\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/lrk-1_tm1898_7days/20241102_053036/FX01898_8X2_f96h20C_600s31x10s10s_A1102da.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/lrk-1_tm1898_7days/20241102_071453/FX01898_8X2_f96h20C_600s31x10s10s_A1102dd.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/lrk-1_tm1898_7days/20241102_130656/FX01898_8x2_f72h20C_600s31x10s10s_B1102db.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/lrk-1_tm1898_7days/20241102_133821/FX01898_8X2_f96h20C_600s31x10s10s_C1102dc.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/lrk-1_tm1898_7days/20241102_144708/FX01898_8x2_f72h20C_600s31x10s10s_B1102de.trv\n",
      "this strain/treatment has 155 total taps\n",
      "this strain/treatment has 5 plates\n",
      "now working on strain lrk-1_tm1898_96hrs\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/lrk-1_tm1898_96hrs/20241102_062418/FX01898_8X2_f96h20C_600s31x10s10s_A1102bc.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/lrk-1_tm1898_96hrs/20241102_121159/FX01898_8x2_f72h20C_600s31x10s10s_B1102ba.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/lrk-1_tm1898_96hrs/20241102_124447/FX01898_8X2_f96h20C_600s31x10s10s_C1102bb.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/lrk-1_tm1898_96hrs/20241102_135948/FX01898_8x2_f72h20C_600s31x10s10s_B1102bd.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov2_2024/lrk-1_tm1898_96hrs/20241102_142908/FX01898_8X2_f96h20C_600s31x10s10s_C1102be.trv\n",
      "this strain/treatment has 155 total taps\n",
      "this strain/treatment has 5 plates\n",
      "now working on strain pdr-1_gk448_7days\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov23_2024/pdr-1_gk448_7days/20241123_052341/VC1024_8x2_f96h20C_600s31x10s10s_A1123cb.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov23_2024/pdr-1_gk448_7days/20241123_070956/VC1024_8x2_f96h20C_600s31x10s10s_A1123ce.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov23_2024/pdr-1_gk448_7days/20241123_114550/VC1024_8X2_f96h20C_600s31x10s10s_C1123ca.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov23_2024/pdr-1_gk448_7days/20241123_125854/VC1024_8x2_f72h20C_600s31x10s10s_B1123cc.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov23_2024/pdr-1_gk448_7days/20241123_133018/VC1024_8X2_f96h20C_600s31x10s10s_C1123cd.trv\n",
      "this strain/treatment has 155 total taps\n",
      "this strain/treatment has 4 plates\n",
      "now working on strain pdr-1_gk448_96hrs\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov23_2024/pdr-1_gk448_96hrs/20241123_045654/VC1024_8x2_f96h20C_600s31x10s10s_A1123da.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov23_2024/pdr-1_gk448_96hrs/20241123_064437/VC1024_8x2_f96h20C_600s31x10s10s_A1123dd.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov23_2024/pdr-1_gk448_96hrs/20241123_123222/VC1024_8x2_f72h20C_600s31x10s10s_B1123db.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov23_2024/pdr-1_gk448_96hrs/20241123_141831/VC1024_8x2_f72h20C_600s31x10s10s_B1123de.trv\n",
      "this strain/treatment has 124 total taps\n",
      "this strain/treatment has 5 plates\n",
      "now working on strain pink-1_ok3538_7days\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/pink-1_ok3538_7days/20241019_064642/RB2547_11x2_f96h20C_600s31x10s10s_A1019da.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/pink-1_ok3538_7days/20241019_082644/RB2547_11x2_f96h20C_600s31x10s10s_A1019dd.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/pink-1_ok3538_7days/20241019_141958/RB2547_11x2_f72h20C_600s31x10s10s_B1019db.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/pink-1_ok3538_7days/20241019_145104/RB2547_11X2_f96h20C_600s31x10s10s_C1019dc.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/pink-1_ok3538_7days/20241019_160023/RB2547_11x2_f72h20C_600s31x10s10s_B1019de.trv\n",
      "this strain/treatment has 155 total taps\n",
      "this strain/treatment has 5 plates\n",
      "now working on strain pink-1_ok3538_96hrs\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/pink-1_ok3538_96hrs/20241019_071026/RB2547_10x2_f96h20C_600s31x10s10s_A1019cb.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/pink-1_ok3538_96hrs/20241019_085134/RB2547_10x2_f96h20C_600s31x10s10s_A1019ce.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/pink-1_ok3538_96hrs/20241019_133403/RB2547_11X2_f96h20C_600s31x10s10s_C1019ca.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/pink-1_ok3538_96hrs/20241019_144556/RB2547_10x2_f72h20C_600s31x10s10s_B1019cc.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Oct19_2024/pink-1_ok3538_96hrs/20241019_151514/RB2547_10X2_f96h20C_600s31x10s10s_C1019cd.trv\n",
      "this strain/treatment has 154 total taps\n",
      "this strain/treatment has 5 plates\n",
      "now working on strain pink-1_tm1779_7days\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/pink-1_tm1779_7days/20241109_015026/FX01779_10x2_f96h20C_600s31x10s10s_A1109da.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/pink-1_tm1779_7days/20241109_033200/FX01779_10x2_f96h20C_600s31x10s10s_A1109dd.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/pink-1_tm1779_7days/20241109_092547/FX01779_10x2_f72h20C_600s31x10s10s_B1109db.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/pink-1_tm1779_7days/20241109_095633/FX01779_10X2_f96h20C_600s31x10s10s_C1109dc.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/pink-1_tm1779_7days/20241109_110412/FX01779_10x2_f72h20C_600s31x10s10s_B1109de.trv\n",
      "this strain/treatment has 155 total taps\n",
      "this strain/treatment has 5 plates\n",
      "now working on strain pink-1_tm1779_96hrs\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/pink-1_tm1779_96hrs/20241109_021642/FX01779_10x2_f96h20C_600s31x10s10s_A1109cb.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/pink-1_tm1779_96hrs/20241109_035518/FX01779_10x2_f96h20C_600s31x10s10s_A1109ce.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/pink-1_tm1779_96hrs/20241109_083854/FX01779_10X2_f96h20C_600s31x10s10s_C1109ca.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/pink-1_tm1779_96hrs/20241109_095136/FX01779_10x2_f72h20C_600s31x10s10s_B1109cc.trv\n",
      "working on file /Volumes/JosephDISK/Kristen-aging/PD_Age_TabHab_Nov9_2024/pink-1_tm1779_96hrs/20241109_102106/FX01779_10X2_f96h20C_600s31x10s10s_C1109cd.trv\n",
      "this strain/treatment has 155 total taps\n",
      "done step 6.2\n"
     ]
    }
   ],
   "source": [
    "DataLists = [0]  #<- generates empty list. 0 is there to account for python's index starting at 0. \n",
    "# we want indexing to start at 1 (when I say #1 I want the first point, not the second point)\n",
    "\n",
    "for s in list(StrainNames.values()): #<- goes through the dictionary in step 6.1 and processes data\n",
    "    if not s == '':\n",
    "#         threading.Thread(target=DataLists.append(ProcessData(s)['Final'])).start()\n",
    "        DataLists.append(ProcessData(s)['Final']) #<- appends all data into a list of dataframes\n",
    "\n",
    "\n",
    "for df in DataLists[1:]: \n",
    "    assign_taps(df, tolerances)\n",
    "for df in DataLists[1:]:    \n",
    "    insert_plates(df)\n",
    "\n",
    "# print(x)\n",
    "# print(DataLists[0])\n",
    "# print(DataLists[1])\n",
    "#print(DataLists[2])\n",
    "#print(len(DataLists))        \n",
    "print('done step 6.2')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# DataLists.append(ProcessData('vps-35_ok1880')['Final'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         time  dura   dist      prob     speed  plate      Date Plate_id  \\\n",
      "0     599.975  1.97  0.486  0.880000  0.246701      1  20250315  A0315dd   \n",
      "1     609.951  1.50  0.341  0.812500  0.227333      1  20250315  A0315dd   \n",
      "2     619.969  1.35  0.329  0.764706  0.243704      1  20250315  A0315dd   \n",
      "3     629.976  1.03  0.239  0.708333  0.232039      1  20250315  A0315dd   \n",
      "4     639.980  0.60  0.133  0.684211  0.221667      1  20250315  A0315dd   \n",
      "..        ...   ...    ...       ...       ...    ...       ...      ...   \n",
      "150   859.958  0.91  0.161  0.387755  0.176923      5  20250315  B0315db   \n",
      "151   869.964  0.81  0.173  0.403509  0.213580      5  20250315  B0315db   \n",
      "152   879.959  0.43  0.079  0.365079  0.183721      5  20250315  B0315db   \n",
      "153   889.990  0.67  0.142  0.280702  0.211940      5  20250315  B0315db   \n",
      "154  1189.965  3.13  0.678  0.565217  0.216613      5  20250315  B0315db   \n",
      "\n",
      "                       Screen  taps  \n",
      "0    PD_Age_TabHab_Mar15_2025   1.0  \n",
      "1    PD_Age_TabHab_Mar15_2025   2.0  \n",
      "2    PD_Age_TabHab_Mar15_2025   3.0  \n",
      "3    PD_Age_TabHab_Mar15_2025   4.0  \n",
      "4    PD_Age_TabHab_Mar15_2025   5.0  \n",
      "..                        ...   ...  \n",
      "150  PD_Age_TabHab_Mar15_2025  27.0  \n",
      "151  PD_Age_TabHab_Mar15_2025  28.0  \n",
      "152  PD_Age_TabHab_Mar15_2025  29.0  \n",
      "153  PD_Age_TabHab_Mar15_2025  30.0  \n",
      "154  PD_Age_TabHab_Mar15_2025  31.0  \n",
      "\n",
      "[155 rows x 10 columns]\n"
     ]
    }
   ],
   "source": [
    "print(DataLists[1])\n",
    "# print(StrainNames.get(1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26\n",
      "26\n",
      "27\n"
     ]
    }
   ],
   "source": [
    "print(len(genotypes))\n",
    "print(len(np.unique(genotypes)))\n",
    "print(len(DataLists))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 7. Grouping Data and Naming (Optional: Add input here)\n",
    "\n",
    "Here, you get to name your data groups/strain! Name your groups however you like under between the quotation marks for each strain.\n",
    "\n",
    "For example: If your Strain1 is N2 and you wish for the group to be called N2,\n",
    "your line should look like:\n",
    "\n",
    "DataLists[x].assign(dataset = 'N2')\n",
    "\n",
    "## Go back to step 6.1 to check which strain is which item on the DataLists.\n",
    "In this example, the first item on DataLists is N2.\n",
    "\n",
    "\n",
    "## Remember: Put your name in quotes. (ex: 'N2' and not N2)\n",
    "\n",
    "As default, the names are set to the unique identifier labels.\n",
    "\n",
    "## Depending on the number of strains you are running the comparison, you may have to delete/add lines of code (following the same format). \n",
    "## Remember to add/delete commas too.\n",
    "\n",
    "# If you want to change your groups, you do that here. \n",
    "For example, if you have 5 strains in your folder but only want to compare between 2 or 3 strains, designate that here and follow through with steps 6 and 7. Once you are done, come back to step 6 and change your groups again (You are going to have to change your graph titles for the second run-through though)!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         time  dura   dist      prob     speed  plate      Date Plate_id  \\\n",
      "0     599.975  1.97  0.486  0.880000  0.246701      1  20250315  A0315dd   \n",
      "1     609.951  1.50  0.341  0.812500  0.227333      1  20250315  A0315dd   \n",
      "2     619.969  1.35  0.329  0.764706  0.243704      1  20250315  A0315dd   \n",
      "3     629.976  1.03  0.239  0.708333  0.232039      1  20250315  A0315dd   \n",
      "4     639.980  0.60  0.133  0.684211  0.221667      1  20250315  A0315dd   \n",
      "..        ...   ...    ...       ...       ...    ...       ...      ...   \n",
      "150   859.982  0.97  0.216  0.312500  0.222680      5  20241109  C1109cd   \n",
      "151   869.982  1.07  0.247  0.347826  0.230841      5  20241109  C1109cd   \n",
      "152   879.981  1.08  0.224  0.372093  0.207407      5  20241109  C1109cd   \n",
      "153   889.980  0.72  0.159  0.232558  0.220833      5  20241109  C1109cd   \n",
      "154  1189.974  1.80  0.476  0.848485  0.264444      5  20241109  C1109cd   \n",
      "\n",
      "                       Screen  taps              dataset  \n",
      "0    PD_Age_TabHab_Mar15_2025   1.0   LRRK2_G2019S_7days  \n",
      "1    PD_Age_TabHab_Mar15_2025   2.0   LRRK2_G2019S_7days  \n",
      "2    PD_Age_TabHab_Mar15_2025   3.0   LRRK2_G2019S_7days  \n",
      "3    PD_Age_TabHab_Mar15_2025   4.0   LRRK2_G2019S_7days  \n",
      "4    PD_Age_TabHab_Mar15_2025   5.0   LRRK2_G2019S_7days  \n",
      "..                        ...   ...                  ...  \n",
      "150   PD_Age_TabHab_Nov9_2024  27.0  pink-1_tm1779_96hrs  \n",
      "151   PD_Age_TabHab_Nov9_2024  28.0  pink-1_tm1779_96hrs  \n",
      "152   PD_Age_TabHab_Nov9_2024  29.0  pink-1_tm1779_96hrs  \n",
      "153   PD_Age_TabHab_Nov9_2024  30.0  pink-1_tm1779_96hrs  \n",
      "154   PD_Age_TabHab_Nov9_2024  31.0  pink-1_tm1779_96hrs  \n",
      "\n",
      "[5695 rows x 11 columns]\n",
      "done step 7\n"
     ]
    }
   ],
   "source": [
    "TotalConcatenated=pd.DataFrame()\n",
    "for d in range(1,len(np.unique(genotypes))+1):\n",
    "    # print(d)\n",
    "    TotalConcatenated=pd.concat([TotalConcatenated,\n",
    "                                 DataLists[d].assign(dataset=StrainNames.get(d))])\n",
    "\n",
    "\n",
    "\n",
    "# TotalConcatenated = pd.concat([ #<- this function joins your data together with an extra column for which group \n",
    "#                           DataLists[1].assign(dataset = \"Wildtype (N2)\"), \n",
    "#                           DataLists[2].assign(dataset = \"cat-2 (e1112)\"),\n",
    "#                           DataLists[3].assign(dataset = \"cat-2 (n4547)\"),\n",
    "#                           DataLists[4].assign(dataset = \"cat-2 (tm2261)\"),\n",
    "#                           DataLists[5].assign(dataset = \"dat-1 (ok157)\"),\n",
    "#                           DataLists[6].assign(dataset = \"cat-1 (ok441)\"),\n",
    "#                           DataLists[7].assign(dataset = \"pdr-1 (gk448)\"),\n",
    "#                           DataLists[8].assign(dataset = \"pdr-1 (tm395)\"),\n",
    "#                           DataLists[9].assign(dataset = \"pdr-1 (tm598)\"),\n",
    "#                           DataLists[10].assign(dataset = \"dop-1 (vs101)\"),\n",
    "#                           DataLists[11].assign(dataset = \"dop-3 (ok295)\"),\n",
    "#                           DataLists[12].assign(dataset = \"snb-5 (ok1434)\"),\n",
    "#                           DataLists[13].assign(dataset = \"unc-10 (ad591)\"),\n",
    "# #                           DataLists[14].assign(dataset = \"GOA1 Mutant (n3055)\"),\n",
    "# #                           DataLists[5].assign(dataset = StrainNames.get(\"Strain_5\")),\n",
    "# #                         DataLists[5].assign(dataset = \"StrainNameOnePoster\"), #<----- example of custom name\n",
    "                          \n",
    "# ])\n",
    "# # ...etc etc\n",
    "TotalConcatenated.reset_index(inplace=False)\n",
    "print(TotalConcatenated)\n",
    "\n",
    "#if TotalConcatenated[\"taps\"].loc[ind] is not 1:\n",
    "#   TotalConcatenated[\"taps\"].loc[ind:indices[c+1]] = list(range(1,len(TotalConcatenated[\"taps\"].loc[ind:indices[c+1]])+1))\n",
    "# missing_taps(TotalConcatenated, accurate_taps, tolerances)\n",
    "\n",
    "print('done step 7')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Setting Colour Palette - Only run the below cell ONCE\n",
    "\n",
    "The following code sets the colour palette for the whole experiment - and then designate one colour to each strain. After this, if as you are graphing you take away some strains, you can do so with the colours still matching accordingly."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         time  dura   dist      prob     speed  plate      Date Plate_id  \\\n",
      "0     599.975  1.97  0.486  0.880000  0.246701      1  20250315  A0315dd   \n",
      "1     609.951  1.50  0.341  0.812500  0.227333      1  20250315  A0315dd   \n",
      "2     619.969  1.35  0.329  0.764706  0.243704      1  20250315  A0315dd   \n",
      "3     629.976  1.03  0.239  0.708333  0.232039      1  20250315  A0315dd   \n",
      "4     639.980  0.60  0.133  0.684211  0.221667      1  20250315  A0315dd   \n",
      "..        ...   ...    ...       ...       ...    ...       ...      ...   \n",
      "150   859.982  0.97  0.216  0.312500  0.222680      5  20241109  C1109cd   \n",
      "151   869.982  1.07  0.247  0.347826  0.230841      5  20241109  C1109cd   \n",
      "152   879.981  1.08  0.224  0.372093  0.207407      5  20241109  C1109cd   \n",
      "153   889.980  0.72  0.159  0.232558  0.220833      5  20241109  C1109cd   \n",
      "154  1189.974  1.80  0.476  0.848485  0.264444      5  20241109  C1109cd   \n",
      "\n",
      "       Screen  taps              dataset  \n",
      "0    PD_Aging   1.0   LRRK2_G2019S_7days  \n",
      "1    PD_Aging   2.0   LRRK2_G2019S_7days  \n",
      "2    PD_Aging   3.0   LRRK2_G2019S_7days  \n",
      "3    PD_Aging   4.0   LRRK2_G2019S_7days  \n",
      "4    PD_Aging   5.0   LRRK2_G2019S_7days  \n",
      "..        ...   ...                  ...  \n",
      "150  PD_Aging  27.0  pink-1_tm1779_96hrs  \n",
      "151  PD_Aging  28.0  pink-1_tm1779_96hrs  \n",
      "152  PD_Aging  29.0  pink-1_tm1779_96hrs  \n",
      "153  PD_Aging  30.0  pink-1_tm1779_96hrs  \n",
      "154  PD_Aging  31.0  pink-1_tm1779_96hrs  \n",
      "\n",
      "[5695 rows x 11 columns]\n"
     ]
    }
   ],
   "source": [
    "# print(TotalConcatenated['dataset'].str.split(\"_\", n=1, expand=True))\n",
    "TotalConcatenated['Screen']='PD_Screen'\n",
    "# TotalConcatenated['Screen']='PD_Aging'\n",
    "print(TotalConcatenated)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         time  dura   dist      prob     speed  plate      Date Plate_id  \\\n",
      "0     599.975  1.97  0.486  0.880000  0.246701      1  20250315  A0315dd   \n",
      "1     609.951  1.50  0.341  0.812500  0.227333      1  20250315  A0315dd   \n",
      "2     619.969  1.35  0.329  0.764706  0.243704      1  20250315  A0315dd   \n",
      "3     629.976  1.03  0.239  0.708333  0.232039      1  20250315  A0315dd   \n",
      "4     639.980  0.60  0.133  0.684211  0.221667      1  20250315  A0315dd   \n",
      "..        ...   ...    ...       ...       ...    ...       ...      ...   \n",
      "150   859.982  0.97  0.216  0.312500  0.222680      5  20241109  C1109cd   \n",
      "151   869.982  1.07  0.247  0.347826  0.230841      5  20241109  C1109cd   \n",
      "152   879.981  1.08  0.224  0.372093  0.207407      5  20241109  C1109cd   \n",
      "153   889.980  0.72  0.159  0.232558  0.220833      5  20241109  C1109cd   \n",
      "154  1189.974  1.80  0.476  0.848485  0.264444      5  20241109  C1109cd   \n",
      "\n",
      "       Screen  taps              dataset    Gene  Allele    Age  \n",
      "0    PD_Aging   1.0   LRRK2_G2019S_7days   LRRK2  G2019S  7days  \n",
      "1    PD_Aging   2.0   LRRK2_G2019S_7days   LRRK2  G2019S  7days  \n",
      "2    PD_Aging   3.0   LRRK2_G2019S_7days   LRRK2  G2019S  7days  \n",
      "3    PD_Aging   4.0   LRRK2_G2019S_7days   LRRK2  G2019S  7days  \n",
      "4    PD_Aging   5.0   LRRK2_G2019S_7days   LRRK2  G2019S  7days  \n",
      "..        ...   ...                  ...     ...     ...    ...  \n",
      "150  PD_Aging  27.0  pink-1_tm1779_96hrs  pink-1  tm1779  96hrs  \n",
      "151  PD_Aging  28.0  pink-1_tm1779_96hrs  pink-1  tm1779  96hrs  \n",
      "152  PD_Aging  29.0  pink-1_tm1779_96hrs  pink-1  tm1779  96hrs  \n",
      "153  PD_Aging  30.0  pink-1_tm1779_96hrs  pink-1  tm1779  96hrs  \n",
      "154  PD_Aging  31.0  pink-1_tm1779_96hrs  pink-1  tm1779  96hrs  \n",
      "\n",
      "[5695 rows x 14 columns]\n"
     ]
    }
   ],
   "source": [
    "TotalConcatenated[['Gene', 'Allele']] = TotalConcatenated['dataset'].str.split('_', n=1, expand=True)\n",
    "# TotalConcatenated[['Gene', 'Allele','Age']] = TotalConcatenated['dataset'].str.split('_', n=0, expand=True)\n",
    "print(TotalConcatenated)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "      time  dura   dist      prob     speed  plate      Date Plate_id  \\\n",
      "0  599.975  1.97  0.486  0.880000  0.246701      1  20250315  A0315dd   \n",
      "1  609.951  1.50  0.341  0.812500  0.227333      1  20250315  A0315dd   \n",
      "2  619.969  1.35  0.329  0.764706  0.243704      1  20250315  A0315dd   \n",
      "3  629.976  1.03  0.239  0.708333  0.232039      1  20250315  A0315dd   \n",
      "4  639.980  0.60  0.133  0.684211  0.221667      1  20250315  A0315dd   \n",
      "\n",
      "     Screen  taps             dataset   Gene  Allele    Age  \n",
      "0  PD_Aging   1.0  LRRK2_G2019S_7days  LRRK2  G2019S  7days  \n",
      "1  PD_Aging   2.0  LRRK2_G2019S_7days  LRRK2  G2019S  7days  \n",
      "2  PD_Aging   3.0  LRRK2_G2019S_7days  LRRK2  G2019S  7days  \n",
      "3  PD_Aging   4.0  LRRK2_G2019S_7days  LRRK2  G2019S  7days  \n",
      "4  PD_Aging   5.0  LRRK2_G2019S_7days  LRRK2  G2019S  7days  \n"
     ]
    }
   ],
   "source": [
    "print(TotalConcatenated.head(5))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "TotalConcatenated['Allele']=TotalConcatenated['Allele'].fillna('N2')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "          time  dura   dist      prob     speed  plate      Date Plate_id  \\\n",
      "0      599.961  2.24  0.806  0.800000  0.359821      1  20240706  A0707bb   \n",
      "1      609.979  0.50  0.135  0.772727  0.270000      1  20240706  A0707bb   \n",
      "2      619.976  0.82  0.285  0.575758  0.347561      1  20240706  A0707bb   \n",
      "3      629.973  0.82  0.251  0.533333  0.306098      1  20240706  A0707bb   \n",
      "4      639.987  0.84  0.268  0.567568  0.319048      1  20240706  A0707bb   \n",
      "...        ...   ...    ...       ...       ...    ...       ...      ...   \n",
      "1820   859.975  1.25  0.261  0.166667  0.208800     59  20241019  B1019ae   \n",
      "1821   869.936  1.42  0.260  0.192308  0.183099     59  20241019  B1019ae   \n",
      "1822   879.975  1.35  0.251  0.240000  0.185926     59  20241019  B1019ae   \n",
      "1823   889.978  1.01  0.197  0.233333  0.195050     59  20241019  B1019ae   \n",
      "1824  1189.941  1.78  0.451  0.760000  0.253371     59  20241019  B1019ae   \n",
      "\n",
      "        Screen  taps      dataset Gene Allele    Age  \n",
      "0     PD_Aging   1.0  N2_N2_7days   N2     N2  7days  \n",
      "1     PD_Aging   2.0  N2_N2_7days   N2     N2  7days  \n",
      "2     PD_Aging   3.0  N2_N2_7days   N2     N2  7days  \n",
      "3     PD_Aging   4.0  N2_N2_7days   N2     N2  7days  \n",
      "4     PD_Aging   5.0  N2_N2_7days   N2     N2  7days  \n",
      "...        ...   ...          ...  ...    ...    ...  \n",
      "1820  PD_Aging  27.0  N2_N2_96hrs   N2     N2  96hrs  \n",
      "1821  PD_Aging  28.0  N2_N2_96hrs   N2     N2  96hrs  \n",
      "1822  PD_Aging  29.0  N2_N2_96hrs   N2     N2  96hrs  \n",
      "1823  PD_Aging  30.0  N2_N2_96hrs   N2     N2  96hrs  \n",
      "1824  PD_Aging  31.0  N2_N2_96hrs   N2     N2  96hrs  \n",
      "\n",
      "[3621 rows x 14 columns]\n"
     ]
    }
   ],
   "source": [
    "print(TotalConcatenated[TotalConcatenated['Gene']=='N2'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "TotalConcatenated=TotalConcatenated.dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "TotalConcatenated.to_csv('tap_output.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# A debugging cell to test for strain 'XJ1' (which is the old N2)\n",
    "print(TotalConcatenated[TotalConcatenated['Allele']=='XJ1'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# If you want the default colour palette #\n",
    "strains = TotalConcatenated['dataset'].unique()\n",
    "pal = dict(zip(strains, sns.color_palette(\"deep\", n_colors=len(strains))))\n",
    "print('palette is:' + str(pal))\n",
    "\n",
    "\n",
    "# pal = {'Wildtype (N2)': 'black', \n",
    "#        'cat-2 (e1112)': 'lavender', \n",
    "#        'cat-2 (n4547)': 'cornflowerblue', \n",
    "#        'cat-2 (tm2261)': 'blue', \n",
    "#        'dat-1 (ok157)': 'blueviolet', \n",
    "#        'cat-1 (ok441)': 'magenta', \n",
    "#        'pdr-1 (gk448)': 'rosybrown', \n",
    "#        'pdr-1 (tm395)': 'lightcoral', \n",
    "#        'pdr-1 (tm598)': 'indianred', \n",
    "#        'dop-1 (vs101)': 'slategray', \n",
    "#        'dop-3 (ok295)': 'darkslategray', \n",
    "#        'snb-5 (ok1434)': 'mediumseagreen', \n",
    "#        'unc-10 (ad591)': 'darkorange'}\n",
    "print('palette is:' + str(pal))\n",
    "# IF YOU WANT TO CUSTOMIZE THE COLOR PALETTE OF THE GRAPHS\n",
    "# pal = {\"N2\": (0.2980392156862745, 0.4470588235294118, 0.6901960784313725),\n",
    "#        \"LX2060\": (0.8666666666666667, 0.5176470588235295, 0.3215686274509804)\n",
    "#     }\n",
    "\n",
    "print('done setting colour palette')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Outputs final data onto a spreadsheet\n",
    "TotalConcatenated.to_csv(\"tap_output.csv\")\n",
    "print('done')\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 7.5: BONUS - Graph by plates to check for anomolies across each strain/treatment\n",
    "\n",
    "Here, we will be graphing each strain on their own by their individual plates - this will help us find any anomolies or outliers that we can then exclude!\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "Strain_pal = sns.choose_colorbrewer_palette('sequential')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "StrainNames.get(\"Strain_1\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "StrainConcatenated = TotalConcatenated[TotalConcatenated.dataset==StrainNames.get(\"Strain_1\")] #Change/dictate which strain you are analyzing\n",
    "print(StrainConcatenated)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "StrainConcatenated = TotalConcatenated[\n",
    "    TotalConcatenated.dataset == 'unc-10 (ad591)'] #Change/dictate which strain you are analyzing\n",
    "StrainName = \"unc-10 (ad591)\" #Also change this, for labeling purposes\n",
    "\n",
    "\n",
    "out1 = widgets.Output() #<- this is part of the code that helps display your graphs in tabs\n",
    "out2 = widgets.Output()\n",
    "out3 = widgets.Output()\n",
    "out4 = widgets.Output()\n",
    "\n",
    "tab = widgets.Tab(children = [out1, out2, out3, out4]) #<- more code to support tab display\n",
    "tab.set_title(0, 'Probability')\n",
    "tab.set_title(1, 'Distance')\n",
    "tab.set_title(2, 'Duration')\n",
    "tab.set_title(3, 'Speed')\n",
    "display(tab)\n",
    "\n",
    "plt.rcParams.update({'lines.markeredgewidth': 1})\n",
    "\n",
    "# Probability\n",
    "with out1:\n",
    "    plt.figure(linewidth = 2.5) #<- Make your figure lines THICCCC for clean display (posters)\n",
    "    plt.figure(figsize=(12,10)) #<- set your figure size (width, length)\n",
    "    plt.gca().xaxis.grid(False) #<- gets rid of x-axis markers to make data look clean \n",
    "    ax = sns.pointplot(x=\"taps\", #<- Here we use seaborn as our graphing package. \n",
    "                     y=\"prob\",  \n",
    "                     data = StrainConcatenated,\n",
    "                     hue = 'plate', #<- Here we use the extra column from step 6 to separate by group\n",
    "                     palette = Strain_pal, #<- Change colour palette here if you like\n",
    "                     ci = 95) #<- Confidence interval. 95 = standard error\n",
    "    plt.xlabel(\"Taps\") #<- X-axis title\n",
    "    plt.ylabel(\"Probability\") #<- Y-Axis title\n",
    "    plt.title(f\"{StrainName} Probability of Tap Habituation, {ISI}ISI\") #<- Figure Title\n",
    "    plt.ylim(0,1)\n",
    "    ax.legend(loc = 'upper right', fontsize = '12') #<- location of your legend\n",
    "    plt.savefig(f'{StrainName}_Probability_{ISI}ISI_ByPlate.png', format='png', dpi=450) #<- saves your file to your folder at certain DPI\n",
    "    plt.show()\n",
    "\n",
    "# Distance\n",
    "with out2:\n",
    "    plt.figure(linewidth = 2.5)\n",
    "    plt.figure(figsize=(12,10))\n",
    "    plt.gca().xaxis.grid(False)\n",
    "    ax = sns.pointplot(x=\"taps\", \n",
    "                     y=\"dist\", \n",
    "                     data = StrainConcatenated,\n",
    "                     hue = 'plate',\n",
    "                     palette = Strain_pal,\n",
    "                     ci = 95)\n",
    "    plt.xlabel(\"Taps\", fontsize = '12')\n",
    "    plt.ylabel(\"Distance\", fontsize = '12')\n",
    "    plt.title(f\"{StrainName} Distance of Tap Habituation, {ISI}ISI\", fontsize = '16')\n",
    "    plt.ylim(0,1.5)\n",
    "    ax.legend(loc = 'upper right', fontsize = '12')\n",
    "    plt.savefig(f'{StrainName}_Distance_{ISI}ISI_ByPlate.png', format='png', dpi=450)\n",
    "    plt.show()\n",
    "\n",
    "# Duration\n",
    "with out3:\n",
    "    plt.figure(linewidth = 2.5)\n",
    "    plt.figure(figsize=(12,10))\n",
    "    plt.gca().xaxis.grid(False)\n",
    "    ax = sns.pointplot(x=\"taps\", \n",
    "                     y=\"dura\", \n",
    "                     data = StrainConcatenated,\n",
    "                     hue = 'plate',\n",
    "                     palette = Strain_pal,\n",
    "                     ci = 95)\n",
    "    plt.xlabel(\"Taps\", fontsize = '12')\n",
    "    plt.ylabel(\"Duration\", fontsize = '12')\n",
    "    plt.title(f\"{StrainName} Duration of Tap Habituation, {ISI}ISI\", fontsize = '16')\n",
    "    plt.ylim(0,3)\n",
    "    ax.legend(loc = 'upper right', fontsize = '12')\n",
    "    plt.savefig(f'{StrainName}_Duration_{ISI}ISI_ByPlate.png', format='png', dpi=450)\n",
    "    plt.show()\n",
    "\n",
    "# Speed\n",
    "with out4:\n",
    "    plt.figure(linewidth = 2.5)\n",
    "    plt.figure(figsize=(12,10))\n",
    "    plt.gca().xaxis.grid(False)\n",
    "    ax = sns.pointplot(x=\"taps\", \n",
    "                     y=\"speed\", \n",
    "                     data = StrainConcatenated,\n",
    "                     hue = 'plate',\n",
    "                     palette = Strain_pal,\n",
    "                     ci = 95)\n",
    "    plt.xlabel(\"Taps\", fontsize = '12')\n",
    "    plt.ylabel(\"Speed\", fontsize = '12')\n",
    "    plt.title(f\"{StrainName} Speed of Tap Habituation, {ISI}ISI\", fontsize = '16')\n",
    "    plt.ylim(0,0.5)\n",
    "    ax.legend(loc = 'upper right', fontsize = '12')\n",
    "    plt.savefig(f'{StrainName}_Speed_{ISI}ISI.png_ByPlate', format='png', dpi=450)\n",
    "    plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 8. Graph Data (Probability, Distance, Duration, Speed of Tap Habituation)\n",
    "\n",
    "## Here, feel free to change the graph title and axis labels! Images are automatically saved in your folder with your data!\n",
    "\n",
    "Note: It has been agreed by the lab that distance is quite an outdated measure, and the three top measures we consider are:\n",
    "\n",
    "## probability, duration and speed."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "PD_tap=pd.read_csv('/Users/Joseph/Desktop/NewData/hab_output.csv').iloc[: , 1:]\n",
    "N2=PD_tap[PD_tap.dataset=='N2']\n",
    "mutant=PD_tap[PD_tap.Allele=='js125']\n",
    "data=pd.concat([N2,mutant], ignore_index=True, sort=False)\n",
    "data['Time']=data.time.round(-1)\n",
    "# print(N2)\n",
    "# print(mutant)\n",
    "print(data)\n",
    "data.to_csv('/Users/Joseph/Desktop/NewData/unc_output.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "out1 = widgets.Output() #<- this is part of the code that helps display your graphs in tabs\n",
    "out2 = widgets.Output()\n",
    "out3 = widgets.Output()\n",
    "out4 = widgets.Output()\n",
    "\n",
    "tab = widgets.Tab(children = [out1, out2, out3, out4]) #<- more code to support tab display\n",
    "tab.set_title(0, 'Probability')\n",
    "tab.set_title(1, 'Distance')\n",
    "tab.set_title(2, 'Duration')\n",
    "tab.set_title(3, 'Speed')\n",
    "display(tab)\n",
    "\n",
    "plt.rcParams.update({'lines.markeredgewidth': 1})\n",
    "sns.set_context(\"poster\")\n",
    "# Probability\n",
    "with out1:\n",
    "    plt.figure(linewidth = 2.5) #<- Make your figure lines THICCCC for clean display (posters)\n",
    "    plt.figure(figsize=(12,10)) #<- set your figure size (width, length)\n",
    "    plt.gca().xaxis.grid(False) #<- gets rid of x-axis markers to make data look clean \n",
    "    ax = sns.lineplot(x=\"Time\", #<- Here we use seaborn as our graphing package. \n",
    "                 y=\"prob\",  \n",
    "                 data = data,\n",
    "                 hue = 'Gene', #<- Here we use the extra column from step 6 to separate by group\n",
    "                 palette = 'deep',\n",
    "                 linewidth=3,\n",
    "                 style='Gene',\n",
    "                 dashes=False,\n",
    "                 markers=['o','o'],\n",
    "                 markersize=12,\n",
    "                 err_style='bars',\n",
    "                 errorbar = 'se') #<- Confidence interval. 95 = standard error\n",
    "    plt.xlabel(\"Time (s)\") #<- X-axis title\n",
    "    plt.ylabel(\"Probability\") #<- Y-Axis title\n",
    "    plt.title(\"Probability of Tap Habituation\") #<- Figure Title\n",
    "    plt.ylim(0,1)\n",
    "    plt.xlim(550,None)\n",
    "    plt.xticks(rotation=90)\n",
    "    ax.legend(loc = 'upper right', fontsize = '20') #<- location of your legend\n",
    "    plt.savefig('/Users/Joseph/Desktop/NewData/unc43prob.png', format='png', dpi=900,bbox_inches='tight') #<- saves your file to your folder at certain DPI\n",
    "    plt.show()\n",
    "    \n",
    "# Distance\n",
    "with out2:\n",
    "    plt.figure(linewidth = 2.5)\n",
    "    plt.figure(figsize=(12,10))\n",
    "    plt.gca().xaxis.grid(False)\n",
    "    ax = sns.pointplot(x=\"Time\", \n",
    "                 y=\"dist\", \n",
    "                 data = data,\n",
    "                 hue = 'dataset',\n",
    "                 palette = 'deep',\n",
    "#                  palette = pal,\n",
    "                 errorbar = 'se'\n",
    "                      )\n",
    "    plt.xlabel(\"Time (s)\", fontsize = '12')\n",
    "    plt.ylabel(\"Distance\", fontsize = '12')\n",
    "    plt.title(\"Distance of Tap Habituation\", fontsize = '16')\n",
    "    plt.ylim(0,None)\n",
    "    ax.legend(loc = 'upper right', fontsize = '12')\n",
    "#     plt.savefig(f'Distance_{ISI}ISI.png', format='png', dpi=900)\n",
    "    plt.show()\n",
    "\n",
    "# Duration\n",
    "with out3:\n",
    "    plt.figure(linewidth = 2.5)\n",
    "    plt.figure(figsize=(12,10))\n",
    "    plt.gca().xaxis.grid(False)\n",
    "    ax = sns.pointplot(x=\"Time\", \n",
    "                 y=\"dura\", \n",
    "                 data = data,\n",
    "                 hue = 'dataset',\n",
    "                 palette = 'deep',\n",
    "#                  palette = pal,\n",
    "                 errorbar = 'se'\n",
    "                      )\n",
    "    plt.xlabel(\"Time (s)\", fontsize = '12')\n",
    "    plt.ylabel(\"Duration\", fontsize = '12')\n",
    "    plt.title(\"Duration of Tap Habituation\", fontsize = '16')\n",
    "    plt.ylim(0,None)\n",
    "    ax.legend(loc = 'upper right', fontsize = '12')\n",
    "#     plt.savefig(f'Duration_{ISI}ISI.png', format='png', dpi=900)\n",
    "    plt.show()\n",
    "\n",
    "# Speed\n",
    "with out4:\n",
    "    plt.figure(linewidth = 2.5)\n",
    "    plt.figure(figsize=(12,10))\n",
    "    plt.gca().xaxis.grid(False)\n",
    "    ax = sns.pointplot(x=\"Time\", \n",
    "                 y=\"speed\", \n",
    "                 data = data,\n",
    "                 hue = 'dataset',\n",
    "                 palette = 'deep',\n",
    "#                  palette = pal,\n",
    "                 errorbar = 'se'\n",
    "                      )\n",
    "    plt.xlabel(\"Time (s)\", fontsize = '12')\n",
    "    plt.ylabel(\"Speed\", fontsize = '12')\n",
    "    plt.title(\"Speed of Tap Habituation\", fontsize = '16')\n",
    "    plt.ylim(0,None)\n",
    "    plt.xticks(rotation=90)\n",
    "    ax.legend(loc = 'upper right', fontsize = '12')\n",
    "#     plt.savefig(f'Speed_{ISI}ISI.png', format='png', dpi=900)\n",
    "    plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "out1 = widgets.Output() #<- this is part of the code that helps display your graphs in tabs\n",
    "out2 = widgets.Output()\n",
    "out3 = widgets.Output()\n",
    "out4 = widgets.Output()\n",
    "\n",
    "tab = widgets.Tab(children = [out1, out2, out3, out4]) #<- more code to support tab display\n",
    "tab.set_title(0, 'Probability')\n",
    "tab.set_title(1, 'Distance')\n",
    "tab.set_title(2, 'Duration')\n",
    "tab.set_title(3, 'Speed')\n",
    "display(tab)\n",
    "\n",
    "plt.rcParams.update({'lines.markeredgewidth': 1})\n",
    "sns.set_context(\"notebook\")\n",
    "# Probability\n",
    "with out1:\n",
    "    plt.figure(linewidth = 2.5) #<- Make your figure lines THICCCC for clean display (posters)\n",
    "    plt.figure(figsize=(12,10)) #<- set your figure size (width, length)\n",
    "    plt.gca().xaxis.grid(False) #<- gets rid of x-axis markers to make data look clean \n",
    "    ax = sns.pointplot(x=\"taps\", #<- Here we use seaborn as our graphing package. \n",
    "                 y=\"prob\",  \n",
    "                 data = TotalConcatenated,\n",
    "                 hue = 'dataset', #<- Here we use the extra column from step 6 to separate by group    \n",
    "                 palette = pal, #<- Change colour palette here if you like\n",
    "                 ci = 68) #<- Confidence interval. 95 = standard error\n",
    "    plt.xlabel(\"Taps\") #<- X-axis title\n",
    "    plt.ylabel(\"Probability\") #<- Y-Axis title\n",
    "    plt.title(f\"Probability of Tap Habituation, {ISI}ISI\") #<- Figure Title\n",
    "    plt.ylim(0,1)\n",
    "    ax.legend(loc = 'upper right', fontsize = '12') #<- location of your legend\n",
    "#     plt.savefig(f'Probability_{ISI}ISI.png', format='png', dpi=900) #<- saves your file to your folder at certain DPI\n",
    "    plt.show()\n",
    "\n",
    "# Distance\n",
    "with out2:\n",
    "    plt.figure(linewidth = 2.5)\n",
    "    plt.figure(figsize=(12,10))\n",
    "    plt.gca().xaxis.grid(False)\n",
    "    ax = sns.pointplot(x=\"taps\", \n",
    "                 y=\"dist\", \n",
    "                 data = TotalConcatenated,\n",
    "                 hue = 'dataset',\n",
    "                 palette = pal,\n",
    "                 ci = 68  #ci=68 for sem\n",
    "                      )\n",
    "    plt.xlabel(\"Taps\", fontsize = '12')\n",
    "    plt.ylabel(\"Distance\", fontsize = '12')\n",
    "    plt.title(f\"Distance of Tap Habituation, {ISI}ISI\", fontsize = '16')\n",
    "    plt.ylim(0,None)\n",
    "    ax.legend(loc = 'upper right', fontsize = '12')\n",
    "#     plt.savefig(f'Distance_{ISI}ISI.png', format='png', dpi=900)\n",
    "    plt.show()\n",
    "\n",
    "# Duration\n",
    "with out3:\n",
    "    plt.figure(linewidth = 2.5)\n",
    "    plt.figure(figsize=(12,10))\n",
    "    plt.gca().xaxis.grid(False)\n",
    "    ax = sns.pointplot(x=\"taps\", \n",
    "                 y=\"dura\", \n",
    "                 data = TotalConcatenated,\n",
    "                 hue = 'dataset',\n",
    "                 palette = pal,\n",
    "                 ci = 68  #ci=68 for sem\n",
    "                      )\n",
    "    plt.xlabel(\"Taps\", fontsize = '12')\n",
    "    plt.ylabel(\"Duration\", fontsize = '12')\n",
    "    plt.title(f\"Duration of Tap Habituation, {ISI}ISI\", fontsize = '16')\n",
    "    plt.ylim(0,None)\n",
    "    ax.legend(loc = 'upper right', fontsize = '12')\n",
    "#     plt.savefig(f'Duration_{ISI}ISI.png', format='png', dpi=900)\n",
    "    plt.show()\n",
    "\n",
    "# Speed\n",
    "with out4:\n",
    "    plt.figure(linewidth = 2.5)\n",
    "    plt.figure(figsize=(12,10))\n",
    "    plt.gca().xaxis.grid(False)\n",
    "    ax = sns.pointplot(x=\"taps\", \n",
    "                 y=\"speed\", \n",
    "                 data = TotalConcatenated,\n",
    "                 hue = 'dataset',\n",
    "                 palette = pal,\n",
    "                 ci = 68 #ci=68 for sem\n",
    "                      )\n",
    "    plt.xlabel(\"Taps\", fontsize = '12')\n",
    "    plt.ylabel(\"Speed\", fontsize = '12')\n",
    "    plt.title(f\"Speed of Tap Habituation, {ISI}ISI\", fontsize = '16')\n",
    "    plt.ylim(0,None)\n",
    "    ax.legend(loc = 'upper right', fontsize = '12')\n",
    "#     plt.savefig(f'Speed_{ISI}ISI.png', format='png', dpi=900)\n",
    "    plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Done!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
